--- minecraft/src/net/minecraft/block/state/pattern/BlockStateMatcher.java
+++ src/main/java/net/minecraft/block/state/pattern/BlockStateMatcher.java
@@ -39,11 +39,11 @@
       } else {
          return false;
       }
    }
 
-   protected <T extends Comparable<T>> boolean matches(IBlockState var1, IProperty<T> var2, Predicate<?> var3) {
+   protected <T extends Comparable<T>> boolean matches(IBlockState var1, IProperty<T> var2, Predicate<T> var3) {
       return var3.apply(var1.getValue(var2));
    }
 
    public <V extends Comparable<V>> BlockStateMatcher where(IProperty<V> var1, Predicate<? extends V> var2) {
       if(!this.blockstate.getProperties().contains(var1)) {
