--- minecraft/src/net/minecraft/server/MinecraftServer.java
+++ src/main/java/net/minecraft/server/MinecraftServer.java
@@ -751,17 +751,17 @@
       return "vanilla";
    }
 
    public CrashReport addServerInfoToCrashReport(CrashReport var1) {
       var1.g().a("Profiler Position", new ICrashReportDetail<String>() {
-         public String a() throws Exception {
+         public String call() throws Exception {
             return MinecraftServer.this.methodProfiler.a?MinecraftServer.this.methodProfiler.c():"N/A (disabled)";
          }
       });
       if(this.playerList != null) {
          var1.g().a("Player Count", new ICrashReportDetail<String>() {
-            public String a() {
+            public String call() {
                return MinecraftServer.this.playerList.getPlayerCount() + " / " + MinecraftServer.this.playerList.getMaxPlayers() + "; " + MinecraftServer.this.playerList.v();
             }
          });
       }
 
@@ -786,11 +786,11 @@
          }
 
          return var5;
       } else {
          boolean var7 = !var2.contains(" ");
-         List var8 = this.commandManager.a(var1, var2, var3);
+         List<String> var8 = this.commandManager.a(var1, var2, var3);
          if(!var8.isEmpty()) {
             for(String var10 : var8) {
                if(var7) {
                   var5.add("/" + var10);
                } else {
