--- minecraft/src/net/minecraft/world/gen/structure/ComponentScatteredFeaturePieces.java
+++ src/main/java/net/minecraft/world/gen/structure/ComponentScatteredFeaturePieces.java
@@ -668,15 +668,15 @@
                   var11.a(var1, var5.a(var15), var9);
                }
 
                BlockPos var21 = var5.a(var10.a(var9, new BlockPos(3, -1 - var13 * 3, 5), var9, new BlockPos(3, 5, 7)));
                var12.a(var1, var21, var9);
-               Map var22 = var12.a(var21, var9);
+               Map<BlockPos, String> var22 = var12.a(var21, var9);
 
-               for(Entry var17 : var22.entrySet()) {
+               for(Entry<BlockPos, String> var17 : var22.entrySet()) {
                   if("chest".equals(var17.getValue())) {
-                     BlockPos var18 = (BlockPos)var17.getKey();
+                     BlockPos var18 = var17.getKey();
                      var1.setTypeAndData(var18, Blocks.AIR.getBlockData(), 3);
                      TileEntity var19 = var1.getTileEntity(var18.down());
                      if(var19 instanceof TileEntityChest) {
                         ((TileEntityChest)var19).a(LootTableList.n, var2.nextLong());
                      }
