--- minecraft/src/net/minecraft/world/gen/structure/MapGenStructure.java
+++ src/main/java/net/minecraft/world/gen/structure/MapGenStructure.java
@@ -154,11 +154,11 @@
       }
 
       if(var13 != null) {
          return var13;
       } else {
-         List var20 = this.getCoordList();
+         List<BlockPos> var20 = this.getCoordList();
          if(var20 != null) {
             BlockPos var21 = null;
 
             for(BlockPos var23 : var20) {
                double var24 = var23.distanceSq(var2);
