--- minecraft/src/net/minecraft/world/gen/structure/StructureComponentTemplate.java
+++ src/main/java/net/minecraft/world/gen/structure/StructureComponentTemplate.java
@@ -47,11 +47,11 @@
    }
 
    public boolean addComponentParts(World var1, Random var2, StructureBoundingBox var3) {
       this.placeSettings.setBoundingBox(var3);
       this.template.addBlocksToWorld(var1, this.templatePosition, this.placeSettings);
-      Map var4 = this.template.getDataBlocks(this.templatePosition, this.placeSettings);
+      Map<BlockPos, String> var4 = this.template.getDataBlocks(this.templatePosition, this.placeSettings);
 
       for(BlockPos var6 : var4.keySet()) {
          String var7 = (String)var4.get(var6);
          this.handleDataMarker(var7, var6, var1, var2, var3);
       }
