--- minecraft/src/net/minecraft/world/gen/structure/StructureOceanMonumentPieces.java
+++ src/main/java/net/minecraft/world/gen/structure/StructureOceanMonumentPieces.java
@@ -574,11 +574,11 @@
             this.boundingBox = new StructureBoundingBox(var2, 39, var3, var2 + 58 - 1, 61, var3 + 58 - 1);
          } else {
             this.boundingBox = new StructureBoundingBox(var2, 39, var3, var2 + 58 - 1, 61, var3 + 58 - 1);
          }
 
-         List var6 = this.generateRoomGraph(var1);
+         List<StructureOceanMonumentPieces.v> var6 = this.generateRoomGraph(var1);
          this.sourceRoom.claimed = true;
          this.childPieces.add(new StructureOceanMonumentPieces.p(var5, this.sourceRoom));
          this.childPieces.add(new StructureOceanMonumentPieces.j(var5, this.coreRoom, var1));
          ArrayList var7 = Lists.newArrayList();
          var7.add(new StructureOceanMonumentPieces.b());
@@ -698,11 +698,11 @@
          this.coreRoom.connections[EnumFacing.EAST.getIndex()].connections[EnumFacing.NORTH.getIndex()].claimed = true;
          this.coreRoom.connections[EnumFacing.UP.getIndex()].claimed = true;
          this.coreRoom.connections[EnumFacing.EAST.getIndex()].connections[EnumFacing.UP.getIndex()].claimed = true;
          this.coreRoom.connections[EnumFacing.NORTH.getIndex()].connections[EnumFacing.UP.getIndex()].claimed = true;
          this.coreRoom.connections[EnumFacing.EAST.getIndex()].connections[EnumFacing.NORTH.getIndex()].connections[EnumFacing.UP.getIndex()].claimed = true;
-         ArrayList var30 = Lists.newArrayList();
+         ArrayList<StructureOceanMonumentPieces.v> var30 = Lists.newArrayList();
 
          for(StructureOceanMonumentPieces.v var37 : var2) {
             if(var37 != null) {
                var37.updateOpenings();
                var30.add(var37);
