--- minecraft/src/net/minecraft/world/gen/structure/StructureComponentTemplate.java
+++ src/main/java/net/minecraft/world/gen/structure/StructureComponentTemplate.java
@@ -47,11 +47,11 @@
    }
 
    public boolean a(World var1, Random var2, StructureBoundingBox var3) {
       this.b.a(var3);
       this.a.b(var1, this.c, this.b);
-      Map var4 = this.a.a(this.c, this.b);
+      Map<BlockPos, String> var4 = this.a.a(this.c, this.b);
 
       for(BlockPos var6 : var4.keySet()) {
          String var7 = (String)var4.get(var6);
          this.a(var7, var6, var1, var2, var3);
       }
