--- minecraft/src/net/minecraft/world/gen/structure/StructureEndCityPieces.java	2016-06-05 00:35:17.265165388 -0700
+++ src/main/java/net/minecraft/world/gen/structure/StructureEndCityPieces.java	2016-06-05 20:39:42.560787365 -0700
@@ -208,7 +208,7 @@
       if(var1 > 8) {
          return false;
       } else {
-         ArrayList var6 = Lists.newArrayList();
+         ArrayList<StructureComponent> var6 = Lists.newArrayList();
          if(var0.generate(var1, var2, var3, var6, var5)) {
             boolean var7 = false;
             int var8 = var5.nextInt();
