--- minecraft/src/net/minecraft/entity/ai/attributes/ModifiableAttributeInstance.java
+++ src/main/java/net/minecraft/entity/ai/attributes/ModifiableAttributeInstance.java
@@ -74,11 +74,11 @@
 
    public void b(AttributeModifier var1) {
       if(this.a(var1.a()) != null) {
          throw new IllegalArgumentException("Modifier is already applied on this attribute!");
       } else {
-         Object var2 = (Set)this.d.get(var1.b());
+         Set<AttributeModifier> var2 = this.d.get(var1.b());
          if(var2 == null) {
             var2 = Sets.newHashSet();
             this.d.put(var1.b(), var2);
          }
 
