--- minecraft/src/net/minecraft/stats/StatList.java	2016-06-05 00:35:40.421209773 -0700
+++ src/main/java/net/minecraft/stats/StatList.java	2016-06-05 20:29:13.421181362 -0700
@@ -128,7 +128,7 @@
    }
 
    private static void initCraftableStats() {
-      HashSet var0 = Sets.newHashSet();
+      HashSet<Item> var0 = Sets.newHashSet();
 
       for(IRecipe var2 : CraftingManager.getInstance().getRecipeList()) {
          if(var2.getRecipeOutput() != null) {
