--- minecraft/src/net/minecraft/entity/projectile/EntityPotion.java
+++ src/main/java/net/minecraft/entity/projectile/EntityPotion.java
@@ -80,11 +80,11 @@
 
    protected void a(RayTraceResult var1) {
       if(!this.world.isClientSide) {
          ItemStack var2 = this.getItem();
          PotionType var3 = PotionUtils.c(var2);
-         List var4 = PotionUtils.getEffects(var2);
+         List<PotionEffect> var4 = PotionUtils.getEffects(var2);
          if(var1.type == RayTraceResult.EnumMovingObjectType.BLOCK && var3 == PotionTypes.b && var4.isEmpty()) {
             BlockPos var18 = var1.a().shift(var1.direction);
             this.a(var18);
 
             for(EnumFacing var22 : EnumFacing.EnumDirectionLimit.HORIZONTAL) {
@@ -109,11 +109,11 @@
                   }
 
                   this.world.addEntity(var5);
                } else {
                   AxisAlignedBB var17 = this.getBoundingBox().grow(4.0D, 2.0D, 4.0D);
-                  List var19 = this.world.a(EntityLivingBase.class, var17);
+                  List<EntityLivingBase> var19 = this.world.a(EntityLivingBase.class, var17);
                   if(!var19.isEmpty()) {
                      for(EntityLivingBase var8 : var19) {
                         if(var8.cI()) {
                            double var9 = this.h(var8);
                            if(var9 < 16.0D) {
