--- minecraft/src/net/minecraft/entity/projectile/EntityDragonFireball.java
+++ src/main/java/net/minecraft/entity/projectile/EntityDragonFireball.java
@@ -28,11 +28,11 @@
       EntityFireball.a(var0, "DragonFireball");
    }
 
    protected void a(RayTraceResult var1) {
       if(!this.world.isClientSide) {
-         List var2 = this.world.a(EntityLivingBase.class, this.getBoundingBox().grow(4.0D, 2.0D, 4.0D));
+         List<EntityLivingBase> var2 = this.world.a(EntityLivingBase.class, this.getBoundingBox().grow(4.0D, 2.0D, 4.0D));
          EntityAreaEffectCloud var3 = new EntityAreaEffectCloud(this.world, this.locX, this.locY, this.locZ);
          var3.setSource(this.shooter);
          var3.setParticle(EnumParticleTypes.DRAGON_BREATH);
          var3.setRadius(3.0F);
          var3.setDuration(2400);
