--- minecraft/src/net/minecraft/world/chunk/Chunk.java
+++ src/main/java/net/minecraft/world/chunk/Chunk.java
@@ -372,11 +372,11 @@
             return Blocks.AIR.getBlockData();
          } catch (Throwable var7) {
             CrashReport var5 = CrashReport.a(var7, "Getting block state");
             CrashReportCategory var6 = var5.a("Block being got");
             var6.a("Location", new ICrashReportDetail<String>() {
-               public String a() throws Exception {
+               public String call() throws Exception {
                   return CrashReportCategory.a(var1, var2, var3);
                }
             });
             throw new ReportedException(var5);
          }
@@ -685,11 +685,11 @@
       int var6 = MathHelper.floor((var2.e + 2.0D) / 16.0D);
       var5 = MathHelper.clamp(var5, 0, this.entitySlices.length - 1);
       var6 = MathHelper.clamp(var6, 0, this.entitySlices.length - 1);
 
       for(int var7 = var5; var7 <= var6; ++var7) {
-         for(Entity var9 : this.entitySlices[var7].c(var1)) {
+         for(T var9 : this.entitySlices[var7].c(var1)) {
             if(var9.getBoundingBox().b(var2) && (var4 == null || var4.apply(var9))) {
                var3.add(var9);
             }
          }
       }
