--- minecraft/src/net/minecraft/entity/item/EntityFallingBlock.java	2016-06-05 00:35:35.045199283 -0700
+++ src/main/java/net/minecraft/entity/item/EntityFallingBlock.java	2016-06-05 20:08:58.067281519 -0700
@@ -153,7 +153,7 @@
       if(this.hurtEntities) {
          int var4 = MathHelper.ceiling_float_int(var1 - 1.0F);
          if(var4 > 0) {
-            ArrayList var5 = Lists.newArrayList(this.worldObj.getEntitiesWithinAABBExcludingEntity(this, this.getEntityBoundingBox()));
+            ArrayList<Entity> var5 = Lists.newArrayList(this.worldObj.getEntitiesWithinAABBExcludingEntity(this, this.getEntityBoundingBox()));
             boolean var6 = var3 == Blocks.ANVIL;
             DamageSource var7 = var6?DamageSource.anvil:DamageSource.fallingBlock;
 
