--- minecraft/src/net/minecraft/crash/CrashReport.java
+++ src/main/java/net/minecraft/crash/CrashReport.java
@@ -69,11 +69,11 @@
          }
       });
       this.theReportCategory.setDetail("JVM Flags", new ICrashReportDetail<String>() {
          public String call() {
             RuntimeMXBean var1 = ManagementFactory.getRuntimeMXBean();
-            List var2 = var1.getInputArguments();
+            List<String> var2 = var1.getInputArguments();
             int var3 = 0;
             StringBuilder var4 = new StringBuilder();
 
             for(String var6 : var2) {
                if(var6.startsWith("-X")) {
