--- minecraft/src/net/minecraft/server/AttributeModifiable.java	2016-06-04 07:22:19.593552490 -0700
+++ src/main/java/net/minecraft/server/AttributeModifiable.java	2016-06-04 20:24:54.888496411 -0700
@@ -74,15 +74,15 @@
         if (this.a(attributemodifier.a()) != null) {
             throw new IllegalArgumentException("Modifier is already applied on this attribute!");
         } else {
-            Object object = (Set) this.d.get(attributemodifier.b());
+            Set<AttributeModifier> object = this.d.get(attributemodifier.b());
 
             if (object == null) {
                 object = Sets.newHashSet();
                 this.d.put(attributemodifier.b(), object);
             }
 
-            ((Set) this.c.get(Integer.valueOf(attributemodifier.c()))).add(attributemodifier);
-            ((Set) object).add(attributemodifier);
+            this.c.get(Integer.valueOf(attributemodifier.c())).add(attributemodifier);
+            object.add(attributemodifier);
             this.e.put(attributemodifier.a(), attributemodifier);
             this.f();
         }
