--- minecraft/src/net/minecraft/entity/ai/attributes/AttributeMap.java	2016-06-04 22:55:01.789727571 -0700
+++ src/main/java/net/minecraft/entity/ai/attributes/AttributeMap.java	2016-06-04 23:57:59.362669179 -0700
@@ -74,14 +74,4 @@
 
       return var1;
    }
-
-   // $FF: synthetic method
-   public IAttributeInstance getAttributeInstanceByName(String var1) {
-      return this.getAttributeInstanceByName(var1);
-   }
-
-   // $FF: synthetic method
-   public IAttributeInstance getAttributeInstance(IAttribute var1) {
-      return this.getAttributeInstance(var1);
-   }
 }
