--- minecraft/src/net/minecraft/server/BaseBlockPosition.java	2016-06-04 07:22:43.705622776 -0700
+++ src/main/java/net/minecraft/server/BaseBlockPosition.java	2016-06-04 09:11:50.448605387 -0700
@@ -89,7 +89,7 @@
         return Objects.toStringHelper(this).add("x", this.getX()).add("y", this.getY()).add("z", this.getZ()).toString();
     }
 
-    public int compareTo(Object object) {
-        return this.l((BaseBlockPosition) object);
+    public int compareTo(BaseBlockPosition object) { // Fountain: Fix decompiler error
+        return this.l(object);
     }
 }
