--- minecraft/src/net/minecraft/block/BlockRedstoneWire.java	2016-06-05 00:35:11.353154393 -0700
+++ src/main/java/net/minecraft/block/BlockRedstoneWire.java	2016-06-05 11:57:54.550623937 -0700
@@ -122,7 +122,7 @@
 
    private IBlockState updateSurroundingRedstone(World var1, BlockPos var2, IBlockState var3) {
       var3 = this.calculateCurrentChanges(var1, var2, var2, var3);
-      ArrayList var4 = Lists.newArrayList(this.blocksNeedingUpdate);
+      ArrayList<BlockPos> var4 = Lists.newArrayList(this.blocksNeedingUpdate);
       this.blocksNeedingUpdate.clear();
 
       for(BlockPos var6 : var4) {
