--- minecraft/src/net/minecraft/block/state/pattern/BlockPattern.java	2016-06-04 22:55:00.789727742 -0700
+++ src/main/java/net/minecraft/block/state/pattern/BlockPattern.java	2016-06-04 23:19:58.580445173 -0700
@@ -110,11 +110,6 @@
       public BlockWorldState load(BlockPos var1) throws Exception {
          return new BlockWorldState(this.world, var1, this.forceLoad);
       }
-
-      // $FF: synthetic method
-      public Object load(Object var1) throws Exception {
-         return this.load((BlockPos)var1);
-      }
    }
 
    public static class b {
