--- minecraft/src/net/minecraft/block/BlockNewLeaf.java	2016-06-04 22:54:43.285731670 -0700
+++ src/main/java/net/minecraft/block/BlockNewLeaf.java	2016-06-04 23:30:31.986138658 -0700
@@ -22,11 +22,6 @@
       public boolean apply(@Nullable BlockPlanks.a var1) {
          return var1.getMetadata() >= 4;
       }
-
-      // $FF: synthetic method
-      public boolean apply(Object var1) {
-         return this.apply((BlockPlanks.a)var1);
-      }
    });
 
    public BlockNewLeaf() {
