--- minecraft/src/net/minecraft/block/BlockTorch.java	2016-06-04 22:54:46.309730863 -0700
+++ src/main/java/net/minecraft/block/BlockTorch.java	2016-06-04 23:33:09.122566778 -0700
@@ -25,11 +25,6 @@
       public boolean apply(@Nullable EnumFacing var1) {
          return var1 != EnumFacing.DOWN;
       }
-
-      // $FF: synthetic method
-      public boolean apply(Object var1) {
-         return this.apply((EnumFacing)var1);
-      }
    });
    protected static final AxisAlignedBB STANDING_AABB = new AxisAlignedBB(0.4000000059604645D, 0.0D, 0.4000000059604645D, 0.6000000238418579D, 0.6000000238418579D, 0.6000000238418579D);
    protected static final AxisAlignedBB TORCH_NORTH_AABB = new AxisAlignedBB(0.3499999940395355D, 0.20000000298023224D, 0.699999988079071D, 0.6499999761581421D, 0.800000011920929D, 1.0D);
