--- minecraft/src/net/minecraft/block/BlockFlower.java	2016-06-04 22:54:46.693730764 -0700
+++ src/main/java/net/minecraft/block/BlockFlower.java	2016-06-04 23:30:17.310098776 -0700
@@ -36,11 +36,6 @@
             public boolean apply(@Nullable BlockFlower.a var1) {
                return var1.getBlockType() == BlockFlower.this.getBlockType();
             }
-
-            // $FF: synthetic method
-            public boolean apply(Object var1) {
-               return this.apply((BlockFlower.a)var1);
-            }
          });
       }
 
@@ -131,11 +126,6 @@
                public boolean apply(@Nullable BlockFlower.a var1) {
                   return var1.getBlockType() == var3;
                }
-
-               // $FF: synthetic method
-               public boolean apply(Object var1) {
-                  return this.apply((BlockFlower.a)var1);
-               }
             });
             TYPES_FOR_BLOCK[var3.ordinal()] = (BlockFlower.a[])var4.toArray(new BlockFlower.a[var4.size()]);
          }
