--- minecraft/src/net/minecraft/block/BlockRailPowered.java	2016-06-04 22:54:42.809731802 -0700
+++ src/main/java/net/minecraft/block/BlockRailPowered.java	2016-06-04 23:31:31.862301566 -0700
@@ -19,11 +19,6 @@
       public boolean apply(@Nullable BlockRailBase.b var1) {
          return var1 != BlockRailBase.b.NORTH_EAST && var1 != BlockRailBase.b.NORTH_WEST && var1 != BlockRailBase.b.SOUTH_EAST && var1 != BlockRailBase.b.SOUTH_WEST;
       }
-
-      // $FF: synthetic method
-      public boolean apply(Object var1) {
-         return this.apply((BlockRailBase.b)var1);
-      }
    });
    public static final PropertyBool POWERED = PropertyBool.create("powered");
 
