--- minecraft/src/net/minecraft/block/BlockSponge.java	2016-06-04 22:54:45.525731068 -0700
+++ src/main/java/net/minecraft/block/BlockSponge.java	2016-06-04 23:32:41.310490870 -0700
@@ -53,7 +53,7 @@
 
    private boolean absorb(World var1, BlockPos var2) {
       LinkedList var3 = Lists.newLinkedList();
-      ArrayList var4 = Lists.newArrayList();
+      ArrayList<BlockPos> var4 = Lists.newArrayList();
       var3.add(new Tuple(var2, Integer.valueOf(0)));
       int var5 = 0;
 
