--- minecraft/src/net/minecraft/server/Block.java	2016-06-04 07:22:17.165545421 -0700
+++ src/main/java/net/minecraft/server/Block.java	2016-06-04 20:27:46.009031021 -0700
@@ -35,7 +35,7 @@
     private String name;
 
     public static int getId(Block block) {
-        return Block.REGISTRY.a((Object) block);
+        return Block.REGISTRY.a(block); // Fountain: Fix decompiler error
     }
 
     public static int getCombinedId(IBlockData iblockdata) {
@@ -863,7 +863,7 @@
 
             if (hashset.contains(block16)) {
                 for (int i = 0; i < 15; ++i) {
-                    int j = Block.REGISTRY.a((Object) block16) << 4 | i;
+                    int j = Block.REGISTRY.a(block16) << 4 | i; // Fountain: Fix decompiler error
 
                     Block.REGISTRY_ID.a(block16.fromLegacyData(i), j);
                 }
@@ -872,7 +872,7 @@
 
                 while (iterator2.hasNext()) {
                     IBlockData iblockdata = (IBlockData) iterator2.next();
-                    int k = Block.REGISTRY.a((Object) block16) << 4 | block16.toLegacyData(iblockdata);
+                    int k = Block.REGISTRY.a(block16) << 4 | block16.toLegacyData(iblockdata); // Fountain: Fix decompiler error
 
                     Block.REGISTRY_ID.a(iblockdata, k);
                 }
