--- minecraft/src/net/minecraft/block/BlockNewLog.java	2016-06-04 22:54:46.393730841 -0700
+++ src/main/java/net/minecraft/block/BlockNewLog.java	2016-06-04 23:30:37.274153033 -0700
@@ -17,11 +17,6 @@
       public boolean apply(@Nullable BlockPlanks.a var1) {
          return var1.getMetadata() >= 4;
       }
-
-      // $FF: synthetic method
-      public boolean apply(Object var1) {
-         return this.apply((BlockPlanks.a)var1);
-      }
    });
 
    public BlockNewLog() {
