--- minecraft/src/net/minecraft/block/BlockPumpkin.java	2016-06-04 22:54:46.373730846 -0700
+++ src/main/java/net/minecraft/block/BlockPumpkin.java	2016-06-04 23:31:15.958258266 -0700
@@ -33,11 +33,6 @@
       public boolean apply(@Nullable IBlockState var1) {
          return var1 != null && (var1.getBlock() == Blocks.PUMPKIN || var1.getBlock() == Blocks.LIT_PUMPKIN);
       }
-
-      // $FF: synthetic method
-      public boolean apply(Object var1) {
-         return this.apply((IBlockState)var1);
-      }
    };
 
    protected BlockPumpkin() {
