--- minecraft/src/net/minecraft/server/BlockStatePredicate.java	2016-06-04 07:22:40.141612377 -0700
+++ src/main/java/net/minecraft/server/BlockStatePredicate.java	2016-06-04 20:35:27.202574600 -0700
@@ -49,7 +49,7 @@
         }
     }
 
-    protected <T extends Comparable<T>> boolean a(IBlockData iblockdata, IBlockState<T> iblockstate, Predicate<?> predicate) {
+    protected <T extends Comparable<T>> boolean a(IBlockData iblockdata, IBlockState<T> iblockstate, Predicate<T> predicate) { // Fountain: Fix decompiler error
         return predicate.apply(iblockdata.get(iblockstate));
     }
 
@@ -62,7 +62,7 @@
         }
     }
 
-    public boolean apply(Object object) {
-        return this.a((IBlockData) object);
+    public boolean apply(IBlockData object) { // Fountain: Fix decompiler error
+        return this.a(object);
     }
 }
