--- minecraft/src/net/minecraft/server/ChatTypeAdapterFactory.java	2016-06-04 07:22:32.925591332 -0700
+++ src/main/java/net/minecraft/server/ChatTypeAdapterFactory.java	2016-06-04 20:37:41.715201141 -0700
@@ -33,7 +33,7 @@
             }
 
             return new TypeAdapter() {
-                public void write(JsonWriter jsonwriter, T t0) throws IOException {
+                public void write(JsonWriter jsonwriter, Object t0) throws IOException { // Fountain: Fix decompiler error
                     if (t0 == null) {
                         jsonwriter.nullValue();
                     } else {
@@ -47,7 +47,7 @@
                         jsonreader.nextNull();
                         return null;
                     } else {
-                        return hashmap.get(jsonreader.nextString());
+                        return (T) hashmap.get(jsonreader.nextString());
                     }
                 }
             };
