--- minecraft/src/net/minecraft/command/CommandHandler.java	2016-06-04 22:54:59.657727943 -0700
+++ src/main/java/net/minecraft/command/CommandHandler.java	2016-06-04 23:40:08.763716957 -0700
@@ -52,7 +52,7 @@
          var1.addChatMessage(var8);
       } else if(var5.checkPermission(this.getServer(), var1)) {
          if(var6 > -1) {
-            List var14 = EntitySelector.matchEntities(var1, var3[var6], Entity.class);
+            List<Entity> var14 = EntitySelector.matchEntities(var1, var3[var6], Entity.class);
             String var9 = var3[var6];
             var1.setCommandStat(CommandResultStats.a.AFFECTED_ENTITIES, var14.size());
 
