--- minecraft/src/net/minecraft/server/CommandAbstract.java	2016-06-04 07:22:25.505569709 -0700
+++ src/main/java/net/minecraft/server/CommandAbstract.java	2016-06-04 20:40:18.627887037 -0700
@@ -212,7 +212,7 @@
         }
 
         if (object != null && oclass.isAssignableFrom(object.getClass())) {
-            return (Entity) object;
+            return (T) object; // Fountain: Fix decompiler error
         } else {
             throw new ExceptionEntityNotFound();
         }
@@ -554,8 +554,8 @@
         return this.getCommand().compareTo(icommand.getCommand());
     }
 
-    public int compareTo(Object object) {
-        return this.a((ICommand) object);
+    public int compareTo(ICommand object) { // Fountain: Fix decompiler error
+        return this.a(object);
     }
 
     public static class CommandNumber {
