--- minecraft/src/net/minecraft/server/Container.java	2016-06-04 07:22:22.285560329 -0700
+++ src/main/java/net/minecraft/server/Container.java	2016-06-04 09:22:22.873323527 -0700
@@ -24,7 +24,7 @@
     protected Slot a(Slot slot) {
         slot.rawSlotIndex = this.c.size();
         this.c.add(slot);
-        this.b.add((Object) null);
+        this.b.add(null); // Fountain: Fix decompiler error
         return slot;
     }
 
