--- minecraft/src/net/minecraft/crash/CrashReportCategory.java	2016-06-04 22:54:50.645729801 -0700
+++ src/main/java/net/minecraft/crash/CrashReportCategory.java	2016-06-04 23:55:01.946179236 -0700
@@ -160,11 +160,6 @@
                return "ID #" + var4;
             }
          }
-
-         // $FF: synthetic method
-         public Object call() throws Exception {
-            return this.call();
-         }
       });
       var0.setDetail("Block data value", new ICrashReportDetail<String>() {
          public String call() throws Exception {
@@ -175,21 +170,11 @@
                return String.format("%1$d / 0x%1$X / 0b%2$s", new Object[]{Integer.valueOf(var3), var1});
             }
          }
-
-         // $FF: synthetic method
-         public Object call() throws Exception {
-            return this.call();
-         }
       });
       var0.setDetail("Block location", new ICrashReportDetail<String>() {
          public String call() throws Exception {
             return CrashReportCategory.getCoordinateInfo(var1);
          }
-
-         // $FF: synthetic method
-         public Object call() throws Exception {
-            return this.call();
-         }
       });
    }
 
@@ -198,21 +183,11 @@
          public String call() throws Exception {
             return var2.toString();
          }
-
-         // $FF: synthetic method
-         public Object call() throws Exception {
-            return this.call();
-         }
       });
       var0.setDetail("Block location", new ICrashReportDetail<String>() {
          public String call() throws Exception {
             return CrashReportCategory.getCoordinateInfo(var1);
          }
-
-         // $FF: synthetic method
-         public Object call() throws Exception {
-            return this.call();
-         }
       });
    }
 
