--- minecraft/src/net/minecraft/server/CustomWorldSettingsFinal.java	2016-06-04 07:22:37.489604641 -0700
+++ src/main/java/net/minecraft/server/CustomWorldSettingsFinal.java	2016-06-04 20:43:09.540593280 -0700
@@ -366,12 +366,12 @@
             return jsonobject;
         }
 
-        public Object deserialize(JsonElement jsonelement, Type type, JsonDeserializationContext jsondeserializationcontext) throws JsonParseException {
+        public CustomWorldSettingsFinal.CustomWorldSettings deserialize(JsonElement jsonelement, Type type, JsonDeserializationContext jsondeserializationcontext) throws JsonParseException { // Fountain: Fix decompiler error
             return this.a(jsonelement, type, jsondeserializationcontext);
         }
 
-        public JsonElement serialize(Object object, Type type, JsonSerializationContext jsonserializationcontext) {
-            return this.a((CustomWorldSettingsFinal.CustomWorldSettings) object, type, jsonserializationcontext);
+        public JsonElement serialize(CustomWorldSettingsFinal.CustomWorldSettings object, Type type, JsonSerializationContext jsonserializationcontext) { // Fountain: Fix decompiler error
+            return this.a(object, type, jsonserializationcontext);
         }
     }
 
