--- minecraft/src/net/minecraft/server/DataConverterManager.java	2016-06-04 07:22:29.229580559 -0700
+++ src/main/java/net/minecraft/server/DataConverterManager.java	2016-06-04 20:43:55.272776616 -0700
@@ -91,7 +91,7 @@
     }
 
     private <V> List<V> a(Map<DataConverterType, List<V>> map, DataConverterType dataconvertertype) {
-        Object object = (List) map.get(dataconvertertype);
+        List<V> object = map.get(dataconvertertype); // Fountain: Fix decompiler error
 
         if (object == null) {
             object = Lists.newArrayList();
