--- minecraft/src/net/minecraft/server/DedicatedPlayerList.java	2016-06-04 07:22:42.341618796 -0700
+++ src/main/java/net/minecraft/server/DedicatedPlayerList.java	2016-06-04 08:47:05.518035612 -0700
@@ -146,7 +146,8 @@
         return this.getOPs().b(gameprofile);
     }
 
+    /* Fountain: Fix decompiler error
     public MinecraftServer getServer() {
         return this.getServer();
-    }
+    }*/
 }
