--- minecraft/src/net/minecraft/server/DispenserRegistry.java	2016-06-04 07:22:37.133603604 -0700
+++ src/main/java/net/minecraft/server/DispenserRegistry.java	2016-06-04 20:54:51.967254303 -0700
@@ -69,7 +69,7 @@
         BlockDispenser.REGISTRY.a(Items.SPLASH_POTION, new IDispenseBehavior() {
             public ItemStack a(ISourceBlock isourceblock, final ItemStack itemstack) {
                 return (new DispenseBehaviorProjectile() {
-                    protected IProjectile a(World world, IPosition iposition, ItemStack itemstack) {
+                    protected IProjectile a(World world, IPosition iposition, ItemStack itemstack1) { // Fountain: Fix decompiler error
                         return new EntityPotion(world, iposition.getX(), iposition.getY(), iposition.getZ(), itemstack1.cloneItemStack());
                     }
 
@@ -86,7 +86,7 @@
         BlockDispenser.REGISTRY.a(Items.LINGERING_POTION, new IDispenseBehavior() {
             public ItemStack a(ISourceBlock isourceblock, final ItemStack itemstack) {
                 return (new DispenseBehaviorProjectile() {
-                    protected IProjectile a(World world, IPosition iposition, ItemStack itemstack) {
+                    protected IProjectile a(World world, IPosition iposition, ItemStack itemstack1) { // Fountain: Fix decompiler error
                         return new EntityPotion(world, iposition.getX(), iposition.getY(), iposition.getZ(), itemstack1.cloneItemStack());
                     }
 
