--- minecraft/src/net/minecraft/server/DragonControllerManager.java	2016-06-04 07:22:17.261545700 -0700
+++ src/main/java/net/minecraft/server/DragonControllerManager.java	2016-06-04 20:55:24.115370968 -0700
@@ -42,6 +42,6 @@
             this.dragonControllers[i] = dragoncontrollerphase.a(this.enderDragon);
         }
 
-        return this.dragonControllers[i];
+        return (T) this.dragonControllers[i]; // Fountain: Fix decompiler error
     }
 }
