--- minecraft/src/net/minecraft/enchantment/EnchantmentHelper.java	2016-06-04 22:55:08.813726534 -0700
+++ src/main/java/net/minecraft/enchantment/EnchantmentHelper.java	2016-06-04 23:56:03.038347929 -0700
@@ -56,7 +56,7 @@
    }
 
    public static Map<Enchantment, Integer> getEnchantments(ItemStack var0) {
-      LinkedHashMap var1 = Maps.newLinkedHashMap();
+      LinkedHashMap<Enchantment, Integer> var1 = Maps.newLinkedHashMap();
       NBTTagList var2 = var0.getItem() == Items.ENCHANTED_BOOK?Items.ENCHANTED_BOOK.getEnchantments(var0):var0.getEnchantmentTagList();
       if(var2 != null) {
          for(int var3 = 0; var3 < var2.tagCount(); ++var3) {
@@ -160,7 +160,7 @@
    }
 
    public static int getMaxEnchantmentLevel(Enchantment var0, EntityLivingBase var1) {
-      Iterable var2 = var0.getEntityEquipment(var1);
+      Iterable<ItemStack> var2 = var0.getEntityEquipment(var1);
       if(var2 == null) {
          return 0;
       } else {
@@ -215,7 +215,7 @@
 
    @Nullable
    public static ItemStack getEnchantedItem(Enchantment var0, EntityLivingBase var1) {
-      Iterable var2 = var0.getEntityEquipment(var1);
+      Iterable<ItemStack> var2 = var0.getEntityEquipment(var1);
       if(var2 == null) {
          return null;
       } else {
@@ -248,7 +248,7 @@
 
    public static ItemStack addRandomEnchantment(Random var0, ItemStack var1, int var2, boolean var3) {
       boolean var4 = var1.getItem() == Items.BOOK;
-      List var5 = buildEnchantmentList(var0, var1, var2, var3);
+      List<EnchantmentData> var5 = buildEnchantmentList(var0, var1, var2, var3);
       if(var4) {
          var1.setItem(Items.ENCHANTED_BOOK);
       }
