--- minecraft/src/net/minecraft/server/EntitySkeleton.java	2016-06-04 07:22:30.193583369 -0700
+++ src/main/java/net/minecraft/server/EntitySkeleton.java	2016-06-04 20:57:26.759813813 -0700
@@ -8,7 +8,7 @@
     private static final DataWatcherObject<Integer> a = DataWatcher.a(EntitySkeleton.class, DataWatcherRegistry.b);
     private static final DataWatcherObject<Boolean> b = DataWatcher.a(EntitySkeleton.class, DataWatcherRegistry.h);
     private final PathfinderGoalBowShoot c = new PathfinderGoalBowShoot(this, 1.0D, 20, 15.0F);
-    private final PathfinderGoalMeleeAttack bw = new PathfinderGoalMeleeAttack(this, 1.2D, flag) {
+    private final PathfinderGoalMeleeAttack bw = new PathfinderGoalMeleeAttack(this, 1.2D, false) { // Fountain: Fix decompiler error
         public void d() {
             super.d();
             EntitySkeleton.this.a(false);
