--- minecraft/src/net/minecraft/entity/ai/EntityAIFindEntityNearest.java	2016-06-04 22:54:40.249732534 -0700
+++ src/main/java/net/minecraft/entity/ai/EntityAIFindEntityNearest.java	2016-06-05 00:01:01.520198226 -0700
@@ -40,18 +40,13 @@
 
             return var1.isInvisible()?false:((double)var1.getDistanceToEntity(EntityAIFindEntityNearest.this.mob) > var2?false:EntityAITarget.isSuitableTarget(EntityAIFindEntityNearest.this.mob, var1, false, true));
          }
-
-         // $FF: synthetic method
-         public boolean apply(Object var1) {
-            return this.apply((EntityLivingBase)var1);
-         }
       };
       this.sorter = new EntityAINearestAttackableTarget.a(var1);
    }
 
    public boolean shouldExecute() {
       double var1 = this.getFollowRange();
-      List var3 = this.mob.worldObj.getEntitiesWithinAABB(this.classToCheck, this.mob.getEntityBoundingBox().expand(var1, 4.0D, var1), this.predicate);
+      List<? extends EntityLivingBase> var3 = this.mob.worldObj.getEntitiesWithinAABB(this.classToCheck, this.mob.getEntityBoundingBox().expand(var1, 4.0D, var1), this.predicate);
       Collections.sort(var3, this.sorter);
       if(var3.isEmpty()) {
          return false;
