--- minecraft/src/net/minecraft/server/Entity.java	2016-06-04 07:22:15.121539472 -0700
+++ src/main/java/net/minecraft/server/Entity.java	2016-06-04 08:59:52.082982813 -0700
@@ -2124,7 +2124,7 @@
         for (Iterator iterator = this.bv().iterator(); iterator.hasNext(); entity.a(oclass, set)) {
             entity = (Entity) iterator.next();
             if (oclass.isAssignableFrom(entity.getClass())) {
-                set.add(entity);
+                set.add((T) entity); // Fountain: Fix decompiler error
             }
         }
 
