--- minecraft/src/net/minecraft/server/EnumDirection.java	2016-06-04 07:22:24.165565805 -0700
+++ src/main/java/net/minecraft/server/EnumDirection.java	2016-06-04 09:14:02.917836761 -0700
@@ -280,8 +280,8 @@
             return Iterators.forArray(this.a());
         }
 
-        public boolean apply(Object object) {
-            return this.a((EnumDirection) object);
+        public boolean apply(EnumDirection object) { // Fountain: Fix decompiler error
+            return this.a(object);
         }
     }
 
