--- minecraft/src/net/minecraft/block/state/pattern/FactoryBlockPattern.java	2016-06-05 00:35:30.149189827 -0700
+++ src/main/java/net/minecraft/block/state/pattern/FactoryBlockPattern.java	2016-06-05 11:47:47.988064744 -0700
@@ -43,7 +43,7 @@
 
                for(char var9 : var5.toCharArray()) {
                   if(!this.symbolMap.containsKey(Character.valueOf(var9))) {
-                     this.symbolMap.put(Character.valueOf(var9), (Object)null);
+                     this.symbolMap.put(Character.valueOf(var9), null);
                   }
                }
             }
