--- minecraft/src/net/minecraft/server/GameProfileBanList.java	2016-06-04 07:22:27.233574742 -0700
+++ src/main/java/net/minecraft/server/GameProfileBanList.java	2016-06-04 21:04:28.133316398 -0700
@@ -52,7 +52,7 @@
         return (GameProfile) gameprofilebanentry.getKey();
     }
 
-    protected String a(Object object) {
-        return this.b((GameProfile) object);
+    protected String a(GameProfile object) { // Fountain: Fix decompiler error
+        return this.b(object);
     }
 }
