--- minecraft/src/net/minecraft/server/IChatBaseComponent.java	2016-06-04 07:22:09.753523853 -0700
+++ src/main/java/net/minecraft/server/IChatBaseComponent.java	2016-06-04 09:21:01.744851416 -0700
@@ -220,11 +220,11 @@
             return (IChatBaseComponent) ChatDeserializer.a(IChatBaseComponent.ChatSerializer.a, s, IChatBaseComponent.class, true);
         }
 
-        public JsonElement serialize(Object object, Type type, JsonSerializationContext jsonserializationcontext) {
-            return this.a((IChatBaseComponent) object, type, jsonserializationcontext);
+        public JsonElement serialize(IChatBaseComponent object, Type type, JsonSerializationContext jsonserializationcontext) { // Fountain: Fix decompiler error
+            return this.a(object, type, jsonserializationcontext);
         }
 
-        public Object deserialize(JsonElement jsonelement, Type type, JsonDeserializationContext jsondeserializationcontext) throws JsonParseException {
+        public IChatBaseComponent deserialize(JsonElement jsonelement, Type type, JsonDeserializationContext jsondeserializationcontext) throws JsonParseException { // Fountain: Fix decompiler error
             return this.a(jsonelement, type, jsondeserializationcontext);
         }
 
