--- minecraft/src/net/minecraft/server/InsensitiveStringMap.java	2016-06-04 07:22:42.517619309 -0700
+++ src/main/java/net/minecraft/server/InsensitiveStringMap.java	2016-06-04 21:07:04.009867519 -0700
@@ -45,7 +45,7 @@
         Iterator iterator = map.entrySet().iterator();
 
         while (iterator.hasNext()) {
-            Entry entry = (Entry) iterator.next();
+            Entry<String, V> entry = (Entry<String, V>) iterator.next(); // Fountain: Fix decompiler error
 
             this.a((String) entry.getKey(), entry.getValue());
         }
@@ -68,7 +68,7 @@
         return this.a.entrySet();
     }
 
-    public Object put(Object object, Object object1) {
-        return this.a((String) object, object1);
+    public Object put(String object, Object object1) { // Fountain: Fix decompiler error
+        return this.a(object, (V) object1);
     }
 }
