--- minecraft/src/net/minecraft/server/IntHashMap.java	2016-06-04 07:22:27.693576083 -0700
+++ src/main/java/net/minecraft/server/IntHashMap.java	2016-06-04 21:08:39.838205534 -0700
@@ -26,7 +26,7 @@
 
         for (IntHashMap.IntHashMapEntry inthashmap_inthashmapentry = this.a[a(j, this.a.length)]; inthashmap_inthashmapentry != null; inthashmap_inthashmapentry = inthashmap_inthashmapentry.c) {
             if (inthashmap_inthashmapentry.a == i) {
-                return inthashmap_inthashmapentry.b;
+                return (V) inthashmap_inthashmapentry.b;
             }
         }
 
@@ -108,7 +108,7 @@
     public V d(int i) {
         IntHashMap.IntHashMapEntry inthashmap_inthashmapentry = this.e(i);
 
-        return inthashmap_inthashmapentry == null ? null : inthashmap_inthashmapentry.b;
+        return inthashmap_inthashmapentry == null ? null : (V) inthashmap_inthashmapentry.b;
     }
 
     @Nullable
