--- minecraft/src/net/minecraft/item/ItemSlab.java	2016-06-05 00:35:20.469171405 -0700
+++ src/main/java/net/minecraft/item/ItemSlab.java	2016-06-05 12:27:43.244308659 -0700
@@ -83,7 +83,7 @@
       return false;
    }
 
-   protected <T extends Comparable<T>> IBlockState makeState(IProperty<T> var1, Comparable<?> var2) {
-      return this.doubleSlab.getDefaultState().withProperty(var1, var2);
+   protected <T extends Comparable<T>> IBlockState makeState(IProperty<T> var1, Comparable<T> var2) {
+      return this.doubleSlab.getDefaultState().withProperty(var1, (T) var2);
    }
 }
