--- minecraft/src/net/minecraft/server/Item.java	2016-06-04 07:22:21.389557719 -0700
+++ src/main/java/net/minecraft/server/Item.java	2016-06-04 08:58:01.018768913 -0700
@@ -34,7 +34,7 @@
     private String name;
 
     public static int getId(Item item) {
-        return item == null ? 0 : Item.REGISTRY.a((Object) item);
+        return item == null ? 0 : Item.REGISTRY.a(item); // Fountain: Fix decompiler error
     }
 
     public static Item getById(int i) {
