--- minecraft/src/net/minecraft/server/LoginListener.java	2016-06-04 07:22:39.617610849 -0700
+++ src/main/java/net/minecraft/server/LoginListener.java	2016-06-04 21:17:21.512978522 -0700
@@ -90,8 +90,8 @@
                         LoginListener.this.networkManager.setCompressionLevel(LoginListener.this.server.aF());
                     }
 
-                    public void operationComplete(Future future) throws Exception {
-                        this.a((ChannelFuture) future);
+                    public void operationComplete(ChannelFuture future) throws Exception { // Fountain: Fix decompiler error
+                        this.a(future);
                     }
                 }, new GenericFutureListener[0]);
             }
