--- minecraft/src/net/minecraft/server/LootItemConditionEntityScore.java	2016-06-04 07:22:18.013547889 -0700
+++ src/main/java/net/minecraft/server/LootItemConditionEntityScore.java	2016-06-04 21:21:44.434161593 -0700
@@ -91,7 +91,7 @@
             return new LootItemConditionEntityScore(linkedhashmap, (LootTableInfo.EntityTarget) ChatDeserializer.a(jsonobject, "entity", jsondeserializationcontext, LootTableInfo.EntityTarget.class));
         }
 
-        public LootItemCondition b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext) {
+        public LootItemConditionEntityScore b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext) {
             return this.a(jsonobject, jsondeserializationcontext);
         }
     }
