--- minecraft/src/net/minecraft/server/LootItemFunctionSmelt.java	2016-06-04 07:22:26.529572692 -0700
+++ src/main/java/net/minecraft/server/LootItemFunctionSmelt.java	2016-06-04 21:26:42.771416553 -0700
@@ -41,7 +41,7 @@
             return new LootItemFunctionSmelt(alootitemcondition);
         }
 
-        public LootItemFunction b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext, LootItemCondition[] alootitemcondition) {
+        public LootItemFunctionSmelt b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext, LootItemCondition[] alootitemcondition) { // Fountain: Fix decompiler error
             return this.a(jsonobject, jsondeserializationcontext, alootitemcondition);
         }
     }
