--- minecraft/src/net/minecraft/server/LootEnchantLevel.java	2016-06-04 07:22:24.785567611 -0700
+++ src/main/java/net/minecraft/server/LootEnchantLevel.java	2016-06-04 21:21:09.182008085 -0700
@@ -39,7 +39,7 @@
             return new LootEnchantLevel(alootitemcondition, lootvaluebounds, flag);
         }
 
-        public LootItemFunction b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext, LootItemCondition[] alootitemcondition) {
+        public LootEnchantLevel b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext, LootItemCondition[] alootitemcondition) { // Fountain: Fix decompiler error
             return this.a(jsonobject, jsondeserializationcontext, alootitemcondition);
         }
     }
