--- minecraft/src/net/minecraft/server/LootItemConditionRandomChanceWithLooting.java	2016-06-04 07:22:35.629599216 -0700
+++ src/main/java/net/minecraft/server/LootItemConditionRandomChanceWithLooting.java	2016-06-04 21:22:21.906323349 -0700
@@ -40,7 +40,7 @@
             return new LootItemConditionRandomChanceWithLooting(ChatDeserializer.l(jsonobject, "chance"), ChatDeserializer.l(jsonobject, "looting_multiplier"));
         }
 
-        public LootItemCondition b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext) {
+        public LootItemConditionRandomChanceWithLooting b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext) { // Fountain: Fix decompiler error
             return this.a(jsonobject, jsondeserializationcontext);
         }
     }
