--- minecraft/src/net/minecraft/server/LootItemFunctionSetTag.java	2016-06-04 07:22:34.525595998 -0700
+++ src/main/java/net/minecraft/server/LootItemFunctionSetTag.java	2016-06-04 21:26:34.499382618 -0700
@@ -48,7 +48,7 @@
             }
         }
 
-        public LootItemFunction b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext, LootItemCondition[] alootitemcondition) {
+        public LootItemFunctionSetTag b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext, LootItemCondition[] alootitemcondition) { // Fountain: Fix decompiler error
             return this.a(jsonobject, jsondeserializationcontext, alootitemcondition);
         }
     }
