--- minecraft/src/net/minecraft/server/LootSelector.java	2016-06-04 07:22:35.733599521 -0700
+++ src/main/java/net/minecraft/server/LootSelector.java	2016-06-04 21:27:07.979519726 -0700
@@ -108,11 +108,11 @@
             return jsonobject;
         }
 
-        public JsonElement serialize(Object object, Type type, JsonSerializationContext jsonserializationcontext) {
-            return this.a((LootSelector) object, type, jsonserializationcontext);
+        public JsonElement serialize(LootSelector object, Type type, JsonSerializationContext jsonserializationcontext) { // Fountain: Fix decompiler error
+            return this.a(object, type, jsonserializationcontext);
         }
 
-        public Object deserialize(JsonElement jsonelement, Type type, JsonDeserializationContext jsondeserializationcontext) throws JsonParseException {
+        public LootSelector deserialize(JsonElement jsonelement, Type type, JsonDeserializationContext jsondeserializationcontext) throws JsonParseException { // Fountain: Fix decompiler error
             return this.a(jsonelement, type, jsondeserializationcontext);
         }
     }
