--- minecraft/src/net/minecraft/server/LootEnchantFunction.java	2016-06-04 07:22:13.573534967 -0700
+++ src/main/java/net/minecraft/server/LootEnchantFunction.java	2016-06-04 21:20:21.925800107 -0700
@@ -46,7 +46,7 @@
             return new LootEnchantFunction(alootitemcondition, (LootValueBounds) ChatDeserializer.a(jsonobject, "count", jsondeserializationcontext, LootValueBounds.class));
         }
 
-        public LootItemFunction b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext, LootItemCondition[] alootitemcondition) {
+        public LootEnchantFunction b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext, LootItemCondition[] alootitemcondition) {
             return this.a(jsonobject, jsondeserializationcontext, alootitemcondition);
         }
     }
