--- minecraft/src/net/minecraft/server/LootItemFunctionSetDamage.java	2016-06-04 07:22:17.761547156 -0700
+++ src/main/java/net/minecraft/server/LootItemFunctionSetDamage.java	2016-06-04 21:26:13.923298027 -0700
@@ -43,7 +43,7 @@
             return new LootItemFunctionSetDamage(alootitemcondition, (LootValueBounds) ChatDeserializer.a(jsonobject, "damage", jsondeserializationcontext, LootValueBounds.class));
         }
 
-        public LootItemFunction b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext, LootItemCondition[] alootitemcondition) {
+        public LootItemFunctionSetDamage b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext, LootItemCondition[] alootitemcondition) { // Fountain: Fix decompiler error
             return this.a(jsonobject, jsondeserializationcontext, alootitemcondition);
         }
     }
