--- minecraft/src/net/minecraft/server/LootItemConditionKilledByPlayer.java	2016-06-04 07:22:19.297551628 -0700
+++ src/main/java/net/minecraft/server/LootItemConditionKilledByPlayer.java	2016-06-04 21:21:57.214216920 -0700
@@ -33,7 +33,7 @@
             return new LootItemConditionKilledByPlayer(ChatDeserializer.a(jsonobject, "inverse", false));
         }
 
-        public LootItemCondition b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext) {
+        public LootItemConditionKilledByPlayer b(JsonObject jsonobject, JsonDeserializationContext jsondeserializationcontext) {
             return this.a(jsonobject, jsondeserializationcontext);
         }
     }
