--- minecraft/src/net/minecraft/server/LootEntityPropertyOnFire.java	2016-06-04 07:22:40.133612354 -0700
+++ src/main/java/net/minecraft/server/LootEntityPropertyOnFire.java	2016-06-04 21:21:21.662062585 -0700
@@ -32,7 +32,7 @@
             return new LootEntityPropertyOnFire(ChatDeserializer.c(jsonelement, "on_fire"));
         }
 
-        public LootEntityProperty a(JsonElement jsonelement, JsonDeserializationContext jsondeserializationcontext) {
+        public LootEntityPropertyOnFire a(JsonElement jsonelement, JsonDeserializationContext jsondeserializationcontext) {
             return this.b(jsonelement, jsondeserializationcontext);
         }
     }
