--- minecraft/src/net/minecraft/nbt/NBTTagCompound.java	2016-06-04 22:55:12.805726069 -0700
+++ src/main/java/net/minecraft/nbt/NBTTagCompound.java	2016-06-04 23:57:24.934574094 -0700
@@ -319,21 +319,11 @@
          public String call() throws Exception {
             return NBTBase.NBT_TYPES[((NBTBase)NBTTagCompound.this.tagMap.get(var1)).getId()];
          }
-
-         // $FF: synthetic method
-         public Object call() throws Exception {
-            return this.call();
-         }
       });
       var5.setDetail("Tag type expected", new ICrashReportDetail<String>() {
          public String call() throws Exception {
             return NBTBase.NBT_TYPES[var2];
          }
-
-         // $FF: synthetic method
-         public Object call() throws Exception {
-            return this.call();
-         }
       });
       var5.addCrashSection("Tag name", var1);
       return var4;
