--- minecraft/src/net/minecraft/server/NetworkManager.java	2016-06-04 07:22:37.581604909 -0700
+++ src/main/java/net/minecraft/server/NetworkManager.java	2016-06-04 21:32:43.784866342 -0700
@@ -114,7 +114,7 @@
     protected void a(ChannelHandlerContext channelhandlercontext, Packet<?> packet) throws Exception {
         if (this.channel.isOpen()) {
             try {
-                packet.a(this.m);
+                packet.a((PacketDataSerializer) this.m); // Fountain: Fix decompiler error
             } catch (CancelledPacketHandleException cancelledpackethandleexception) {
                 ;
             }
@@ -310,8 +310,8 @@
         }
     }
 
-    protected void channelRead0(ChannelHandlerContext channelhandlercontext, Object object) throws Exception {
-        this.a(channelhandlercontext, (Packet) object);
+    protected void channelRead0(ChannelHandlerContext channelhandlercontext, Packet object) throws Exception { // Fountain: Fix decompiler error
+        this.a(channelhandlercontext, object);
     }
 
     static class QueuedPacket {
