--- minecraft/src/net/minecraft/server/NextTickListEntry.java	2016-06-04 07:22:27.553575675 -0700
+++ src/main/java/net/minecraft/server/NextTickListEntry.java	2016-06-04 21:33:05.244951031 -0700
@@ -50,7 +50,7 @@
         return this.e;
     }
 
-    public int compareTo(Object object) {
-        return this.a((NextTickListEntry) object);
+    public int compareTo(NextTickListEntry object) { // Fountain: Fix decompiler error
+        return this.a(object);
     }
 }
