--- minecraft/src/net/minecraft/server/SoundEffects.java	2016-06-04 07:22:36.693602319 -0700
+++ src/main/java/net/minecraft/server/SoundEffects.java	2016-06-04 08:26:13.672781951 -0700
@@ -224,7 +224,7 @@
     public static final SoundEffect dl;
     public static final SoundEffect dm;
     public static final SoundEffect dn;
-    public static final SoundEffect do;
+    public static final SoundEffect doMetalBreak; // Fountain: Fix decompiler error
     public static final SoundEffect dp;
     public static final SoundEffect dq;
     public static final SoundEffect dr;
@@ -683,7 +683,7 @@
             dl = a("entity.magmacube.hurt");
             dm = a("entity.magmacube.jump");
             dn = a("entity.magmacube.squish");
-            do = a("block.metal.break");
+            doMetalBreak = a("block.metal.break"); // Fountain: Fix decompiler error
             dp = a("block.metal.fall");
             dq = a("block.metal.hit");
             dr = a("block.metal.place");
