--- minecraft/src/net/minecraft/server/MinecraftServer.java	2016-06-05 00:34:58.813131539 -0700
+++ src/main/java/net/minecraft/server/MinecraftServer.java	2016-06-11 03:13:47.370670226 -0700
@@ -137,6 +137,12 @@
    private Thread serverThread;
    private long currentTime = getCurrentTimeMillis();
 
+   private static DedicatedServer dedicatedServer = null;
+
+   public static DedicatedServer getDedicatedServer() {
+       return dedicatedServer;
+   }
+
    public MinecraftServer(File var1, Proxy var2, DataFixer var3, YggdrasilAuthenticationService var4, MinecraftSessionService var5, GameProfileRepository var6, PlayerProfileCache var7) {
       this.serverProxy = var2;
       this.authService = var4;
@@ -324,6 +330,8 @@
    }
 
    protected void stopServer() {
+      org.fountainmc.api.Fountain.getServer().getPluginManager()
+              .fireEvent(new org.fountainmc.api.event.server.ServerStopEvent());
       LOG.info("Stopping server");
       if(this.getNetworkSystem() != null) {
          this.getNetworkSystem().terminateEndpoints();
@@ -678,11 +686,14 @@
          }
 
          var12.startServerThread();
+         org.fountainmc.api.Fountain.getServer().getPluginManager()
+                 .fireEvent(new org.fountainmc.api.event.server.ServerStartEvent(var12.getServerPort(), var12.getServerOwner(), var0));
          Runtime.getRuntime().addShutdownHook(new Thread("Server Shutdown Thread") {
             public void run() {
                var12.stopServer();
             }
          });
+         dedicatedServer = var12;
       } catch (Exception var14) {
          LOG.fatal("Failed to start the minecraft server", var14);
       }
@@ -786,7 +797,7 @@
          return var5;
       } else {
          boolean var7 = !var2.contains(" ");
-         List var8 = this.commandManager.getTabCompletionOptions(var1, var2, var3);
+         List<String> var8 = this.commandManager.getTabCompletionOptions(var1, var2, var3);
          if(!var8.isEmpty()) {
             for(String var10 : var8) {
                if(var7) {
