--- minecraft/src/net/minecraft/entity/EntityTracker.java	2016-06-05 00:35:03.909140749 -0700
+++ src/main/java/net/minecraft/entity/EntityTracker.java	2016-06-05 20:16:33.845589412 -0700
@@ -250,8 +250,8 @@
    }
 
    public void sendLeashedEntitiesInChunk(EntityPlayerMP var1, Chunk var2) {
-      ArrayList var3 = Lists.newArrayList();
-      ArrayList var4 = Lists.newArrayList();
+      ArrayList<Entity> var3 = Lists.newArrayList();
+      ArrayList<Entity> var4 = Lists.newArrayList();
 
       for(EntityTrackerEntry var6 : this.trackedEntities) {
          Entity var7 = var6.getTrackedEntity();
