--- minecraft/src/net/minecraft/util/datafix/DataFixer.java
+++ src/main/java/net/minecraft/util/datafix/DataFixer.java
@@ -90,14 +90,14 @@
 
       }
    }
 
    private <V> List<V> a(Map<IFixType, List<V>> var1, IFixType var2) {
-      Object var3 = (List)var1.get(var2);
+      List<V> var3 = var1.get(var2);
       if(var3 == null) {
          var3 = Lists.newArrayList();
          var1.put(var2, var3);
       }
 
-      return (List<V>)var3;
+      return var3;
    }
 }
