--- minecraft/src/net/minecraft/server/dedicated/DedicatedServer.java
+++ src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -302,11 +302,11 @@
 
    protected void systemExitNow() {
       System.exit(0);
    }
 
-   protected void updateTimeLightAndEntities() {
+   public void updateTimeLightAndEntities() {
       super.updateTimeLightAndEntities();
       this.executePendingCommands();
    }
 
    public boolean getAllowNether() {
