--- minecraft/src/net/minecraft/scoreboard/Scoreboard.java
+++ src/main/java/net/minecraft/scoreboard/Scoreboard.java
@@ -40,11 +40,11 @@
          ScoreObjective var3 = this.getObjective(var1);
          if(var3 != null) {
             throw new IllegalArgumentException("An objective with the name \'" + var1 + "\' already exists!");
          } else {
             var3 = new ScoreObjective(this, var1, var2);
-            Object var4 = (List)this.objectivesByCriteria.get(var2);
+            List<ScoreObjective> var4 = this.objectivesByCriteria.get(var2);
             if(var4 == null) {
                var4 = Lists.newArrayList();
                this.objectivesByCriteria.put(var2, var4);
             }
 
@@ -73,11 +73,11 @@
 
    public Score getPlayerScoreForObjective(String var1, ScoreObjective var2) {
       if(var1.length() > 40) {
          throw new IllegalArgumentException("The player name \'" + var1 + "\' is too long!");
       } else {
-         Object var3 = (Map)this.playerScores.get(var1);
+         Map<ScoreObjective, Score> var3 = this.playerScores.get(var1);
          if(var3 == null) {
             var3 = Maps.newHashMap();
             this.playerScores.put(var1, var3);
          }
 
@@ -135,11 +135,11 @@
       }
 
    }
 
    public Collection<Score> getScores() {
-      Collection var1 = this.playerScores.values();
+      Collection<Map<ScoreObjective, Score>> var1 = this.playerScores.values();
       ArrayList var2 = Lists.newArrayList();
 
       for(Map var4 : var1) {
          var2.addAll(var4.values());
       }
