--- minecraft/src/net/minecraft/scoreboard/ServerScoreboard.java
+++ src/main/java/net/minecraft/scoreboard/ServerScoreboard.java
@@ -156,11 +156,11 @@
 
       return var2;
    }
 
    public void e(ScoreObjective var1) {
-      List var2 = this.getScoreboardScorePacketsForObjective(var1);
+      List<Packet<?>> var2 = this.getScoreboardScorePacketsForObjective(var1);
 
       for(EntityPlayerMP var4 : this.a.getPlayerList().v()) {
          for(Packet var6 : var2) {
             var4.playerConnection.sendPacket(var6);
          }
@@ -181,11 +181,11 @@
 
       return var2;
    }
 
    public void g(ScoreObjective var1) {
-      List var2 = this.f(var1);
+      List<Packet<?>> var2 = this.f(var1);
 
       for(EntityPlayerMP var4 : this.a.getPlayerList().v()) {
          for(Packet var6 : var2) {
             var4.playerConnection.sendPacket(var6);
          }
