--- minecraft/src/net/minecraft/scoreboard/ServerScoreboard.java
+++ src/main/java/net/minecraft/scoreboard/ServerScoreboard.java
@@ -156,14 +156,14 @@
 
       return var2;
    }
 
    public void addObjective(ScoreObjective var1) {
-      List var2 = this.getCreatePackets(var1);
+      List<Packet<?>> var2 = this.getCreatePackets(var1);
 
       for(EntityPlayerMP var4 : this.scoreboardMCServer.getPlayerList().getPlayerList()) {
-         for(Packet var6 : var2) {
+         for(Packet<?> var6 : var2) {
             var4.connection.sendPacket(var6);
          }
       }
 
       this.addedObjectives.add(var1);
@@ -181,14 +181,14 @@
 
       return var2;
    }
 
    public void sendDisplaySlotRemovalPackets(ScoreObjective var1) {
-      List var2 = this.getDestroyPackets(var1);
+      List<Packet<?>> var2 = this.getDestroyPackets(var1);
 
       for(EntityPlayerMP var4 : this.scoreboardMCServer.getPlayerList().getPlayerList()) {
-         for(Packet var6 : var2) {
+         for(Packet<?> var6 : var2) {
             var4.connection.sendPacket(var6);
          }
       }
 
       this.addedObjectives.remove(var1);
