--- minecraft/src/net/minecraft/world/end/DragonFightManager.java
+++ src/main/java/net/minecraft/world/end/DragonFightManager.java
@@ -287,11 +287,11 @@
       for(EntityPlayerMP var3 : this.world.getPlayers(EntityPlayerMP.class, VALID_PLAYER)) {
          this.bossInfo.addPlayer(var3);
          var1.add(var3);
       }
 
-      HashSet var5 = Sets.newHashSet(this.bossInfo.getPlayers());
+      HashSet<EntityPlayerMP> var5 = Sets.newHashSet(this.bossInfo.getPlayers());
       var5.removeAll(var1);
 
       for(EntityPlayerMP var4 : var5) {
          this.bossInfo.removePlayer(var4);
       }
