--- minecraft/src/net/minecraft/world/chunk/storage/AnvilChunkLoader.java
+++ src/main/java/net/minecraft/world/chunk/storage/AnvilChunkLoader.java
@@ -227,11 +227,11 @@
          NBTTagCompound var27 = var24.writeToNBT(new NBTTagCompound());
          var19.appendTag(var27);
       }
 
       var3.setTag("TileEntities", var19);
-      List var22 = var2.getPendingBlockUpdates(var1, false);
+      List<NextTickListEntry> var22 = var2.getPendingBlockUpdates(var1, false);
       if(var22 != null) {
          long var25 = var2.getTotalWorldTime();
          NBTTagList var28 = new NBTTagList();
 
          for(NextTickListEntry var30 : var22) {
