--- minecraft/src/net/minecraft/nbt/NBTTagCompound.java
+++ src/main/java/net/minecraft/nbt/NBTTagCompound.java
@@ -315,24 +315,24 @@
 
    private CrashReport a(final String var1, final int var2, ClassCastException var3) {
       CrashReport var4 = CrashReport.a(var3, "Reading NBT data");
       CrashReportCategory var5 = var4.a("Corrupt NBT tag", 1);
       var5.a("Tag type found", new ICrashReportDetail<String>() {
-         public String a() throws Exception {
+         public String call() throws Exception {
             return NBTBase.a[((NBTBase)NBTTagCompound.this.map.get(var1)).getTypeId()];
          }
       });
       var5.a("Tag type expected", new ICrashReportDetail<String>() {
-         public String a() throws Exception {
+         public String call() throws Exception {
             return NBTBase.a[var2];
          }
       });
       var5.a((String)"Tag name", var1);
       return var4;
    }
 
-   public NBTTagCompound g() {
+   public NBTTagCompound clone() {
       NBTTagCompound var1 = new NBTTagCompound();
 
       for(String var3 : this.map.keySet()) {
          var1.set(var3, ((NBTBase)this.map.get(var3)).clone());
       }
