--- minecraft/src/net/minecraft/entity/player/EntityPlayer.java
+++ src/main/java/net/minecraft/entity/player/EntityPlayer.java
@@ -495,11 +495,11 @@
    }
 
    public void b(Entity var1, int var2) {
       if(var1 != this) {
          this.addScore(var2);
-         Collection var3 = this.getScoreboard().getObjectivesForCriteria(IScoreCriteria.f);
+         Collection<ScoreObjective> var3 = this.getScoreboard().getObjectivesForCriteria(IScoreCriteria.f);
          if(var1 instanceof EntityPlayer) {
             this.b(StatList.D);
             var3.addAll(this.getScoreboard().getObjectivesForCriteria(IScoreCriteria.e));
          } else {
             this.b(StatList.B);
