--- minecraft/src/net/minecraft/entity/player/EntityPlayerMP.java
+++ src/main/java/net/minecraft/entity/player/EntityPlayerMP.java
@@ -342,21 +342,21 @@
    }
 
    protected void o() {
       Biome var1 = this.world.getBiome(new BlockPos(MathHelper.floor(this.locX), 0, MathHelper.floor(this.locZ)));
       String var2 = var1.l();
-      JsonSerializableSet var3 = (JsonSerializableSet)this.getStatisticManager().b(AchievementList.L);
+      JsonSerializableSet var3 = (JsonSerializableSet)this.getStatisticManager().b((StatBase)AchievementList.L);
       if(var3 == null) {
          var3 = (JsonSerializableSet)this.getStatisticManager().a(AchievementList.L, new JsonSerializableSet());
       }
 
       var3.add(var2);
       if(this.getStatisticManager().b(AchievementList.L) && var3.size() >= Biome.i.size()) {
-         HashSet var4 = Sets.newHashSet(Biome.i);
+         HashSet<Biome> var4 = Sets.newHashSet(Biome.i);
 
          for(String var6 : var3) {
-            Iterator var7 = var4.iterator();
+            Iterator<Biome> var7 = var4.iterator();
 
             while(var7.hasNext()) {
                Biome var8 = (Biome)var7.next();
                if(var8.l().equals(var6)) {
                   var7.remove();
