--- minecraft/src/net/minecraft/world/end/DragonFightManager.java	2016-06-05 00:35:13.501158372 -0700
+++ src/main/java/net/minecraft/world/end/DragonFightManager.java	2016-06-05 20:36:53.959894866 -0700
@@ -289,7 +289,7 @@
          var1.add(var3);
       }
 
-      HashSet var5 = Sets.newHashSet(this.bossInfo.getPlayers());
+      HashSet<EntityPlayerMP> var5 = Sets.newHashSet(this.bossInfo.getPlayers());
       var5.removeAll(var1);
 
       for(EntityPlayerMP var4 : var5) {
