--- minecraft/src/net/minecraft/world/gen/structure/template/Template.java
+++ src/main/java/net/minecraft/world/gen/structure/template/Template.java
@@ -94,11 +94,11 @@
 
       }
    }
 
    private void takeEntitiesFromWorld(World var1, BlockPos var2, BlockPos var3) {
-      List var4 = var1.getEntitiesWithinAABB(Entity.class, new AxisAlignedBB(var2, var3), new Predicate<Entity>() {
+      List<Entity> var4 = var1.getEntitiesWithinAABB(Entity.class, new AxisAlignedBB(var2, var3), new Predicate<Entity>() {
          public boolean apply(@Nullable Entity var1) {
             return !(var1 instanceof EntityPlayer);
          }
       });
       this.entities.clear();
