--- minecraft/src/net/minecraft/entity/EntityTracker.java
+++ src/main/java/net/minecraft/entity/EntityTracker.java
@@ -248,12 +248,12 @@
       }
 
    }
 
    public void sendLeashedEntitiesInChunk(EntityPlayerMP var1, Chunk var2) {
-      ArrayList var3 = Lists.newArrayList();
-      ArrayList var4 = Lists.newArrayList();
+      ArrayList<Entity> var3 = Lists.newArrayList();
+      ArrayList<Entity> var4 = Lists.newArrayList();
 
       for(EntityTrackerEntry var6 : this.trackedEntities) {
          Entity var7 = var6.getTrackedEntity();
          if(var7 != var1 && var7.chunkCoordX == var2.xPosition && var7.chunkCoordZ == var2.zPosition) {
             var6.updatePlayerEntity(var1);
