--- minecraft/src/net/minecraft/entity/EntityAreaEffectCloud.java
+++ src/main/java/net/minecraft/entity/EntityAreaEffectCloud.java
@@ -218,21 +218,21 @@
                if(this.ticksExisted >= ((Integer)var21.getValue()).intValue()) {
                   var18.remove();
                }
             }
 
-            ArrayList var19 = Lists.newArrayList();
+            ArrayList<PotionEffect> var19 = Lists.newArrayList();
 
             for(PotionEffect var25 : this.potion.getEffects()) {
                var19.add(new PotionEffect(var25.getPotion(), var25.getDuration() / 4, var25.getAmplifier(), var25.getIsAmbient(), var25.doesShowParticles()));
             }
 
             var19.addAll(this.effects);
             if(var19.isEmpty()) {
                this.reapplicationDelayMap.clear();
             } else {
-               List var23 = this.worldObj.getEntitiesWithinAABB(EntityLivingBase.class, this.getEntityBoundingBox());
+               List<EntityLivingBase> var23 = this.worldObj.getEntitiesWithinAABB(EntityLivingBase.class, this.getEntityBoundingBox());
                if(!var23.isEmpty()) {
                   for(EntityLivingBase var28 : var23) {
                      if(!this.reapplicationDelayMap.containsKey(var28) && var28.canBeHitWithPotion()) {
                         double var30 = var28.posX - this.posX;
                         double var33 = var28.posZ - this.posZ;
