--- minecraft/src/net/minecraft/entity/EntityAreaEffectCloud.java
+++ src/main/java/net/minecraft/entity/EntityAreaEffectCloud.java
@@ -245,21 +245,21 @@
                if(this.ticksLived >= ((Integer)var20.getValue()).intValue()) {
                   var17.remove();
                }
             }
 
-            ArrayList var18 = Lists.newArrayList();
+            ArrayList<PotionEffect> var18 = Lists.newArrayList();
 
             for(PotionEffect var24 : this.potionRegistry.a()) {
                var18.add(new PotionEffect(var24.getMobEffect(), var24.getDuration() / 4, var24.getAmplifier(), var24.isAmbient(), var24.isShowParticles()));
             }
 
             var18.addAll(this.effects);
             if(var18.isEmpty()) {
                this.au.clear();
             } else {
-               List var22 = this.world.a(EntityLivingBase.class, this.getBoundingBox());
+               List<EntityLivingBase> var22 = this.world.a(EntityLivingBase.class, this.getBoundingBox());
                if(!var22.isEmpty()) {
                   for(EntityLivingBase var27 : var22) {
                      if(!this.au.containsKey(var27) && var27.cI()) {
                         double var29 = var27.locX - this.locX;
                         double var32 = var27.locZ - this.locZ;
