--- minecraft/src/net/minecraft/entity/Entity.java
+++ src/main/java/net/minecraft/entity/Entity.java
@@ -485,11 +485,11 @@
          for(int var23 = var54.size(); var22 < var23; ++var22) {
             var3 = ((AxisAlignedBB)var54.get(var22)).b(this.getBoundingBox(), var3);
          }
 
          this.a(this.getBoundingBox().c(0.0D, var3, 0.0D));
-         var22 = this.onGround || var15 != var3 && var15 < 0.0D;
+         boolean var22b = this.onGround || var15 != var3 && var15 < 0.0D;
          int var56 = 0;
 
          for(int var24 = var54.size(); var56 < var24; ++var56) {
             var1 = ((AxisAlignedBB)var54.get(var56)).a(this.getBoundingBox(), var1);
          }
@@ -500,11 +500,11 @@
          for(int var60 = var54.size(); var56 < var60; ++var56) {
             var5 = ((AxisAlignedBB)var54.get(var56)).c(this.getBoundingBox(), var5);
          }
 
          this.a(this.getBoundingBox().c(0.0D, 0.0D, var5));
-         if(this.P > 0.0F && var22 && (var13 != var1 || var17 != var5)) {
+         if(this.P > 0.0F && var22b && (var13 != var1 || var17 != var5)) {
             double var58 = var1;
             double var25 = var3;
             double var27 = var5;
             AxisAlignedBB var29 = this.getBoundingBox();
             this.a(var21);
@@ -1823,30 +1823,30 @@
       return false;
    }
 
    public void appendEntityCrashDetails(CrashReportCategory var1) {
       var1.a("Entity Type", new ICrashReportDetail<String>() {
-         public String a() throws Exception {
+         public String call() throws Exception {
             return EntityList.b(Entity.this) + " (" + Entity.this.getClass().getCanonicalName() + ")";
          }
       });
       var1.a((String)"Entity ID", Integer.valueOf(this.id));
       var1.a("Entity Name", new ICrashReportDetail<String>() {
-         public String a() throws Exception {
+         public String call() throws Exception {
             return Entity.this.getName();
          }
       });
       var1.a((String)"Entity\'s Exact location", String.format("%.2f, %.2f, %.2f", new Object[]{Double.valueOf(this.locX), Double.valueOf(this.locY), Double.valueOf(this.locZ)}));
       var1.a((String)"Entity\'s Block location", CrashReportCategory.a(MathHelper.floor(this.locX), MathHelper.floor(this.locY), MathHelper.floor(this.locZ)));
       var1.a((String)"Entity\'s Momentum", String.format("%.2f, %.2f, %.2f", new Object[]{Double.valueOf(this.motX), Double.valueOf(this.motY), Double.valueOf(this.motZ)}));
       var1.a("Entity\'s Passengers", new ICrashReportDetail<String>() {
-         public String a() throws Exception {
+         public String call() throws Exception {
             return Entity.this.bx().toString();
          }
       });
       var1.a("Entity\'s Vehicle", new ICrashReportDetail<String>() {
-         public String a() throws Exception {
+         public String call() throws Exception {
             return Entity.this.bB().toString();
          }
       });
    }
 
@@ -2080,19 +2080,19 @@
       this.a((Class)Entity.class, (Set)var1);
       return var1;
    }
 
    public <T extends Entity> Collection<T> b(Class<T> var1) {
-      HashSet var2 = Sets.newHashSet();
-      this.a((Class)var1, (Set)var2);
+      HashSet<T> var2 = Sets.newHashSet();
+      this.a(var1, var2);
       return var2;
    }
 
    private <T extends Entity> void a(Class<T> var1, Set<T> var2) {
       for(Entity var4 : this.bx()) {
          if(var1.isAssignableFrom(var4.getClass())) {
-            var2.add(var4);
+            var2.add((T) var4);
          }
 
          var4.a(var1, var2);
       }
 
