--- minecraft/src/net/minecraft/entity/EntityLivingBase.java
+++ src/main/java/net/minecraft/entity/EntityLivingBase.java
@@ -1223,11 +1223,10 @@
    protected void handleJumpLava() {
       this.motionY += 0.03999999910593033D;
    }
 
    public void moveEntityWithHeading(float var1, float var2) {
-       System.out.println(" -> " + this.motionX + " " + this.motionY + " " + this.motionZ);
       if(this.isServerWorld() || this.canPassengerSteer()) {
          if(!this.isInWater() || this instanceof EntityPlayer && ((EntityPlayer)this).capabilities.isFlying) {
             if(!this.isInLava() || this instanceof EntityPlayer && ((EntityPlayer)this).capabilities.isFlying) {
                if(this.isElytraFlying()) {
                   if(this.motionY > -0.5D) {
