--- minecraft/src/net/minecraft/entity/EntityTracker.java
+++ src/main/java/net/minecraft/entity/EntityTracker.java
@@ -150,11 +150,11 @@
       } catch (Throwable var10) {
          CrashReport var6 = CrashReport.a(var10, "Adding entity to track");
          CrashReportCategory var7 = var6.a("Entity To Track");
          var7.a((String)"Tracking range", var2 + " blocks");
          var7.a("Update interval", new ICrashReportDetail<String>() {
-            public String a() throws Exception {
+            public String call() throws Exception {
                String var1 = "Once per " + var3 + " ticks";
                if(var3 == Integer.MAX_VALUE) {
                   var1 = "Maximum (" + var1 + ")";
                }
 
@@ -248,12 +248,12 @@
       }
 
    }
 
    public void a(EntityPlayerMP var1, Chunk var2) {
-      ArrayList var3 = Lists.newArrayList();
-      ArrayList var4 = Lists.newArrayList();
+      ArrayList<Entity> var3 = Lists.newArrayList();
+      ArrayList<Entity> var4 = Lists.newArrayList();
 
       for(EntityTrackerEntry var6 : this.c) {
          Entity var7 = var6.b();
          if(var7 != var1 && var7.ac == var2.locX && var7.ae == var2.locZ) {
             var6.updatePlayer(var1);
