--- minecraft/src/net/minecraft/world/storage/loot/conditions/EntityHasProperty.java
+++ src/main/java/net/minecraft/world/storage/loot/conditions/EntityHasProperty.java
@@ -57,15 +57,15 @@
          var1.add("properties", var4);
          var1.add("entity", var3.serialize(var2.target));
       }
 
       public EntityHasProperty deserialize(JsonObject var1, JsonDeserializationContext var2) {
-         Set var3 = JsonUtils.getJsonObject(var1, "properties").entrySet();
+         Set<Entry<String, JsonElement>> var3 = JsonUtils.getJsonObject(var1, "properties").entrySet();
          EntityProperty[] var4 = new EntityProperty[var3.size()];
          int var5 = 0;
 
-         for(Entry var7 : var3) {
+         for(Entry<String, JsonElement> var7 : var3) {
             var4[var5++] = EntityPropertyManager.getSerializerForName(new ResourceLocation((String)var7.getKey())).deserialize((JsonElement)var7.getValue(), var2);
          }
 
          return new EntityHasProperty(var4, (LootContext.b)JsonUtils.deserializeClass(var1, "entity", var2, LootContext.b.class));
       }
