--- minecraft/src/net/minecraft/server/management/PlayerProfileCache.java	2016-06-05 00:35:06.265145043 -0700
+++ src/main/java/net/minecraft/server/management/PlayerProfileCache.java	2016-06-05 12:32:54.769536763 -0700
@@ -181,7 +181,7 @@
 
       try {
          var1 = Files.newReader(this.usercacheFile, Charsets.UTF_8);
-         List var2 = (List)this.gson.fromJson(var1, TYPE);
+         List<PlayerProfileCache.a> var2 = this.gson.fromJson(var1, TYPE);
          this.usernameToProfileEntryMap.clear();
          this.uuidToProfileEntryMap.clear();
          this.gameProfiles.clear();
