--- minecraft/src/net/minecraft/server/management/PlayerInteractionManager.java
+++ src/main/java/net/minecraft/server/management/PlayerInteractionManager.java
@@ -24,10 +24,13 @@
 import net.minecraft.world.ILockableContainer;
 import net.minecraft.world.World;
 import net.minecraft.world.WorldServer;
 import net.minecraft.world.WorldSettings;
 
+import org.fountainmc.api.event.world.BlockBreakEvent;
+import org.fountainmc.api.world.BlockPosition;
+
 public class PlayerInteractionManager {
    public World theWorld;
    public EntityPlayerMP thisPlayerMP;
    private WorldSettings.a gameType = WorldSettings.a.NOT_SET;
    private boolean isDestroyingBlock;
@@ -221,10 +224,22 @@
                      return false;
                   }
                }
             }
 
+            // Fountain start
+            BlockBreakEvent event = BlockBreakEvent.create(new org.fountainmc.entity.WetEntityPlayer(thisPlayerMP), new BlockPosition(theWorld.getFountainWorld(), var1.getX(), var1.getY(), var1.getZ()), theWorld.getBlockState(var1).getFountainState());
+            org.fountainmc.api.Fountain.getServer().getEventManager().fire(event);
+            if (event.isCancelled()) {
+               thisPlayerMP.connection.sendPacket(new SPacketBlockChange(theWorld, var1));
+               TileEntity tileEntity = theWorld.getTileEntity(var1);
+               if (tileEntity != null)
+                   thisPlayerMP.connection.sendPacket(tileEntity.getUpdatePacket());
+               return false;
+            }
+            // Fountain end
+
             this.theWorld.playEvent(this.thisPlayerMP, 2001, var1, Block.getStateId(var2));
             boolean var8 = this.removeBlock(var1);
             if(this.isCreative()) {
                this.thisPlayerMP.connection.sendPacket(new SPacketBlockChange(this.theWorld, var1));
             } else {
@@ -254,11 +269,11 @@
       } else if(var1.getCooldownTracker().hasCooldown(var3.getItem())) {
          return EnumActionResult.PASS;
       } else {
          int var5 = var3.stackSize;
          int var6 = var3.getMetadata();
-         ActionResult var7 = var3.useItemRightClick(var2, var1, var4);
+         ActionResult<ItemStack> var7 = var3.useItemRightClick(var2, var1, var4);
          ItemStack var8 = (ItemStack)var7.getResult();
          if(var8 == var3 && var8.stackSize == var5 && var8.getMaxItemUseDuration() <= 0 && var8.getMetadata() == var6) {
             return var7.getType();
          } else {
             var1.setHeldItem(var4, var8);
