--- minecraft/src/net/minecraft/server/management/PlayerProfileCache.java
+++ src/main/java/net/minecraft/server/management/PlayerProfileCache.java
@@ -179,11 +179,11 @@
    public void load() {
       BufferedReader var1 = null;
 
       try {
          var1 = Files.newReader(this.usercacheFile, Charsets.UTF_8);
-         List var2 = (List)this.gson.fromJson(var1, TYPE);
+         List<PlayerProfileCache.a> var2 = this.gson.fromJson(var1, TYPE);
          this.usernameToProfileEntryMap.clear();
          this.uuidToProfileEntryMap.clear();
          this.gameProfiles.clear();
          if(var2 != null) {
             for(PlayerProfileCache.a var4 : Lists.reverse(var2)) {
