--- minecraft/src/net/minecraft/server/management/PlayerList.java
+++ src/main/java/net/minecraft/server/management/PlayerList.java
@@ -363,11 +363,11 @@
       }
    }
 
    public EntityPlayerMP processLogin(GameProfile var1) {
       UUID var2 = EntityPlayer.a(var1);
-      ArrayList var3 = Lists.newArrayList();
+      ArrayList<EntityPlayerMP> var3 = Lists.newArrayList();
 
       for(int var4 = 0; var4 < this.players.size(); ++var4) {
          EntityPlayerMP var5 = (EntityPlayerMP)this.players.get(var4);
          if(var5.getUniqueID().equals(var2)) {
             var3.add(var5);
