--- minecraft/src/net/minecraft/inventory/ContainerRepair.java
+++ src/main/java/net/minecraft/inventory/ContainerRepair.java
@@ -121,11 +121,11 @@
          this.g.setItem(0, (ItemStack)null);
          this.a = 0;
       } else {
          ItemStack var5 = var1.cloneItemStack();
          ItemStack var6 = this.h.getItem(1);
-         Map var7 = EnchantmentHelper.a(var5);
+         Map<Enchantment, Integer> var7 = EnchantmentHelper.a(var5);
          var3 = var3 + var1.getRepairCost() + (var6 == null?0:var6.getRepairCost());
          this.k = 0;
          if(var6 != null) {
             boolean var8 = var6.getItem() == Items.ENCHANTED_BOOK && !Items.ENCHANTED_BOOK.h(var6).isEmpty();
             if(var5.e() && var5.getItem().a(var1, var6)) {
@@ -166,11 +166,11 @@
                      var5.setData(var13);
                      var2 += 2;
                   }
                }
 
-               Map var19 = EnchantmentHelper.a(var6);
+               Map<Enchantment, Integer> var19 = EnchantmentHelper.a(var6);
 
                for(Enchantment var23 : var19.keySet()) {
                   if(var23 != null) {
                      int var25 = var7.containsKey(var23)?((Integer)var7.get(var23)).intValue():0;
                      int var26 = ((Integer)var19.get(var23)).intValue();
