--- minecraft/src/net/minecraft/inventory/Container.java
+++ src/main/java/net/minecraft/inventory/Container.java
@@ -31,11 +31,11 @@
    }
 
    protected Slot a(Slot var1) {
       var1.rawSlotIndex = this.c.size();
       this.c.add(var1);
-      this.b.add((Object)null);
+      this.b.add(null);
       return var1;
    }
 
    public void addSlotListener(IContainerListener var1) {
       if(this.listeners.contains(var1)) {
