--- minecraft/src/net/minecraft/pathfinding/WalkNodeProcessor.java
+++ src/main/java/net/minecraft/pathfinding/WalkNodeProcessor.java
@@ -61,11 +61,11 @@
       }
 
       BlockPos var9 = new BlockPos(this.b);
       PathNodeType var10 = this.a(this.b, var9.getX(), var1, var9.getZ());
       if(this.b.a(var10) < 0.0F) {
-         HashSet var4 = Sets.newHashSet();
+         HashSet<BlockPos> var4 = Sets.newHashSet();
          var4.add(new BlockPos(this.b.getBoundingBox().a, (double)var1, this.b.getBoundingBox().c));
          var4.add(new BlockPos(this.b.getBoundingBox().a, (double)var1, this.b.getBoundingBox().f));
          var4.add(new BlockPos(this.b.getBoundingBox().d, (double)var1, this.b.getBoundingBox().c));
          var4.add(new BlockPos(this.b.getBoundingBox().d, (double)var1, this.b.getBoundingBox().f));
 
@@ -227,11 +227,11 @@
          }
       }
    }
 
    public PathNodeType a(IBlockAccess var1, int var2, int var3, int var4, EntityLiving var5, int var6, int var7, int var8, boolean var9, boolean var10) {
-      EnumSet var11 = EnumSet.noneOf(PathNodeType.class);
+      EnumSet<PathNodeType> var11 = EnumSet.noneOf(PathNodeType.class);
       PathNodeType var12 = PathNodeType.BLOCKED;
       double var13 = (double)var5.width / 2.0D;
       BlockPos var15 = new BlockPos(var5);
 
       for(int var16 = 0; var16 < var6; ++var16) {
