--- minecraft/src/net/minecraft/entity/ai/EntityAIFollowParent.java
+++ src/main/java/net/minecraft/entity/ai/EntityAIFollowParent.java
@@ -18,11 +18,11 @@
 
    public boolean a() {
       if(this.a.getAge() >= 0) {
          return false;
       } else {
-         List var1 = this.a.world.a(this.a.getClass(), this.a.getBoundingBox().grow(8.0D, 4.0D, 8.0D));
+         List<EntityAnimal> var1 = this.a.world.a(this.a.getClass(), this.a.getBoundingBox().grow(8.0D, 4.0D, 8.0D));
          EntityAnimal var2 = null;
          double var3 = Double.MAX_VALUE;
 
          for(EntityAnimal var6 : var1) {
             if(var6.getAge() >= 0) {
