--- minecraft/src/net/minecraft/entity/ai/EntityAIAvoidEntity.java	2016-06-05 00:35:05.473143597 -0700
+++ src/main/java/net/minecraft/entity/ai/EntityAIAvoidEntity.java	2016-06-05 19:45:28.423153148 -0700
@@ -46,11 +46,11 @@
    }
 
    public boolean shouldExecute() {
-      List var1 = this.theEntity.worldObj.getEntitiesWithinAABB(this.classToAvoid, this.theEntity.getEntityBoundingBox().expand((double)this.avoidDistance, 3.0D, (double)this.avoidDistance), Predicates.and(new Predicate[]{EntitySelectors.CAN_AI_TARGET, this.canBeSeenSelector, this.avoidTargetSelector}));
+      List<T> var1 = this.theEntity.worldObj.getEntitiesWithinAABB(this.classToAvoid, this.theEntity.getEntityBoundingBox().expand((double)this.avoidDistance, 3.0D, (double)this.avoidDistance), Predicates.and(new Predicate[]{EntitySelectors.CAN_AI_TARGET, this.canBeSeenSelector, this.avoidTargetSelector}));
       if(var1.isEmpty()) {
          return false;
       } else {
-         this.closestLivingEntity = (Entity)var1.get(0);
+         this.closestLivingEntity = var1.get(0);
          Vec3d var2 = RandomPositionGenerator.findRandomTargetBlockAwayFrom(this.theEntity, 16, 7, new Vec3d(this.closestLivingEntity.posX, this.closestLivingEntity.posY, this.closestLivingEntity.posZ));
          if(var2 == null) {
             return false;
