--- minecraft/src/net/minecraft/block/state/BlockStateBase.java
+++ src/main/java/net/minecraft/block/state/BlockStateBase.java
@@ -22,20 +22,20 @@
             IProperty var2 = (IProperty)var1.getKey();
             return var2.getName() + "=" + this.getPropertyName(var2, (Comparable)var1.getValue());
          }
       }
 
-      private <T extends Comparable<T>> String getPropertyName(IProperty<T> var1, Comparable<?> var2) {
-         return var1.getName(var2);
+      private <T extends Comparable<T>> String getPropertyName(IProperty<T> var1, Comparable<T> var2) {
+         return var1.getName((T) var2);
       }
    };
 
    public BlockStateBase() {
    }
 
    public <T extends Comparable<T>> IBlockState cycleProperty(IProperty<T> var1) {
-      return this.withProperty(var1, (Comparable)cyclePropertyValue(var1.getAllowedValues(), this.getValue(var1)));
+      return this.withProperty(var1, cyclePropertyValue(var1.getAllowedValues(), this.getValue(var1)));
    }
 
    protected static <T> T cyclePropertyValue(Collection<T> var0, T var1) {
       Iterator var2 = var0.iterator();
 
