--- minecraft/src/net/minecraft/entity/item/EntityFallingBlock.java
+++ src/main/java/net/minecraft/entity/item/EntityFallingBlock.java
@@ -151,11 +151,11 @@
    public void fall(float var1, float var2) {
       Block var3 = this.fallTile.getBlock();
       if(this.hurtEntities) {
          int var4 = MathHelper.ceiling_float_int(var1 - 1.0F);
          if(var4 > 0) {
-            ArrayList var5 = Lists.newArrayList(this.worldObj.getEntitiesWithinAABBExcludingEntity(this, this.getEntityBoundingBox()));
+            ArrayList<Entity> var5 = Lists.newArrayList(this.worldObj.getEntitiesWithinAABBExcludingEntity(this, this.getEntityBoundingBox()));
             boolean var6 = var3 == Blocks.ANVIL;
             DamageSource var7 = var6?DamageSource.anvil:DamageSource.fallingBlock;
 
             for(Entity var9 : var5) {
                var9.attackEntityFrom(var7, (float)Math.min(MathHelper.floor_float((float)var4 * this.fallHurtAmount), this.fallHurtMax));
