--- minecraft/src/net/minecraft/world/gen/FlatGeneratorInfo.java
+++ src/main/java/net/minecraft/world/gen/FlatGeneratorInfo.java
@@ -67,22 +67,22 @@
          var1.append(";");
       } else {
          var1.append(";");
          int var9 = 0;
 
-         for(Entry var4 : this.structures.entrySet()) {
+         for(Entry<String, Map<String, String>> var4 : this.structures.entrySet()) {
             if(var9++ > 0) {
                var1.append(",");
             }
 
             var1.append(((String)var4.getKey()).toLowerCase());
-            Map var5 = (Map)var4.getValue();
+            Map<String, String> var5 = var4.getValue();
             if(!var5.isEmpty()) {
                var1.append("(");
                int var6 = 0;
 
-               for(Entry var8 : var5.entrySet()) {
+               for(Entry<String, String> var8 : var5.entrySet()) {
                   if(var6++ > 0) {
                      var1.append(" ");
                   }
 
                   var1.append((String)var8.getKey());
