--- minecraft/src/net/minecraft/world/World.java
+++ src/main/java/net/minecraft/world/World.java
@@ -68,10 +68,14 @@
 import net.minecraft.world.storage.ISaveHandler;
 import net.minecraft.world.storage.MapStorage;
 import net.minecraft.world.storage.WorldInfo;
 import net.minecraft.world.storage.loot.LootTableManager;
 
+import org.fountainmc.WetServer;
+import org.fountainmc.world.WetWorld;
+import org.fountainmc.world.WorldHookListener;
+
 public abstract class World implements IBlockAccess {
    private int seaLevel = 63;
    protected boolean scheduledUpdatesAreImmediate;
    public final List<Entity> loadedEntityList = Lists.newArrayList();
    protected final List<Entity> unloadedEntityList = Lists.newArrayList();
@@ -109,13 +113,20 @@
    protected boolean spawnHostileMobs;
    protected boolean spawnPeacefulMobs;
    private boolean processingLoadedTiles;
    private final WorldBorder worldBorder;
    int[] lightUpdateBlockList;
+   // Fountain start
+   private final WetWorld fountainWorld = new WetWorld(WetServer.getInstance(), this);
 
+   public WetWorld getFountainWorld() {
+      return fountainWorld;
+   }
+   // Fountain end
+
    protected World(ISaveHandler var1, WorldInfo var2, WorldProvider var3, Profiler var4, boolean var5) {
-      this.eventListeners = Lists.newArrayList(new IWorldEventListener[]{this.pathListener});
+      this.eventListeners = Lists.newArrayList(new IWorldEventListener[]{this.pathListener, new WorldHookListener()});
       this.theCalendar = Calendar.getInstance();
       this.worldScoreboard = new Scoreboard();
       this.spawnHostileMobs = true;
       this.spawnPeacefulMobs = true;
       this.lightUpdateBlockList = new int['\u8000'];
@@ -1982,27 +1993,27 @@
 
       return var4;
    }
 
    public <T extends Entity> List<T> getEntities(Class<? extends T> var1, Predicate<? super T> var2) {
-      ArrayList var3 = Lists.newArrayList();
+      ArrayList<T> var3 = Lists.newArrayList();
 
       for(Entity var5 : this.loadedEntityList) {
-         if(var1.isAssignableFrom(var5.getClass()) && var2.apply(var5)) {
-            var3.add(var5);
+         if(var1.isAssignableFrom(var5.getClass()) && var2.apply((T) var5)) {
+            var3.add((T) var5);
          }
       }
 
       return var3;
    }
 
    public <T extends Entity> List<T> getPlayers(Class<? extends T> var1, Predicate<? super T> var2) {
-      ArrayList var3 = Lists.newArrayList();
+      ArrayList<T> var3 = Lists.newArrayList();
 
       for(Entity var5 : this.playerEntities) {
-         if(var1.isAssignableFrom(var5.getClass()) && var2.apply(var5)) {
-            var3.add(var5);
+         if(var1.isAssignableFrom(var5.getClass()) && var2.apply((T) var5)) {
+            var3.add((T) var5);
          }
       }
 
       return var3;
    }
