--- minecraft/src/net/minecraft/entity/monster/EntityWitch.java
+++ src/main/java/net/minecraft/entity/monster/EntityWitch.java
@@ -102,11 +102,11 @@
             if(this.bx-- <= 0) {
                this.a(false);
                ItemStack var5 = this.getItemInMainHand();
                this.setSlot(EntityEquipmentSlot.MAINHAND, (ItemStack)null);
                if(var5 != null && var5.getItem() == Items.POTION) {
-                  List var6 = PotionUtils.getEffects(var5);
+                  List<PotionEffect> var6 = PotionUtils.getEffects(var5);
                   if(var6 != null) {
                      for(PotionEffect var4 : var6) {
                         this.addEffect(new PotionEffect(var4));
                      }
                   }
