--- minecraft/src/net/minecraft/entity/monster/EntityWitch.java
+++ src/main/java/net/minecraft/entity/monster/EntityWitch.java
@@ -96,11 +96,11 @@
             if(this.witchAttackTimer-- <= 0) {
                this.setAggressive(false);
                ItemStack var5 = this.getHeldItemMainhand();
                this.setItemStackToSlot(EntityEquipmentSlot.MAINHAND, (ItemStack)null);
                if(var5 != null && var5.getItem() == Items.POTIONITEM) {
-                  List var6 = PotionUtils.getEffectsFromStack(var5);
+                  List<PotionEffect> var6 = PotionUtils.getEffectsFromStack(var5);
                   if(var6 != null) {
                      for(PotionEffect var4 : var6) {
                         this.addPotionEffect(new PotionEffect(var4));
                      }
                   }
