--- minecraft/src/net/minecraft/entity/monster/EntitySkeleton.java
+++ src/main/java/net/minecraft/entity/monster/EntitySkeleton.java
@@ -63,11 +63,11 @@
 
 public class EntitySkeleton extends EntityMob implements IRangedAttackMob {
    private static final DataParameter<Integer> a = EntityDataManager.a(EntitySkeleton.class, DataSerializers.b);
    private static final DataParameter<Boolean> b = EntityDataManager.a(EntitySkeleton.class, DataSerializers.h);
    private final EntityAIAttackRangedBow c = new EntityAIAttackRangedBow(this, 1.0D, 20, 15.0F);
-   private final EntityAIAttackMelee bx = new EntityAIAttackMelee(this, 1.2D, var5) {
+   private final EntityAIAttackMelee bx = new EntityAIAttackMelee(this, 1.2D, false) {
       public void d() {
          super.d();
          EntitySkeleton.this.a(false);
       }
 
