--- minecraft/src/net/minecraft/block/BlockSponge.java
+++ src/main/java/net/minecraft/block/BlockSponge.java
@@ -51,11 +51,11 @@
 
    }
 
    private boolean b(World var1, BlockPos var2) {
       LinkedList var3 = Lists.newLinkedList();
-      ArrayList var4 = Lists.newArrayList();
+      ArrayList<BlockPos> var4 = Lists.newArrayList();
       var3.add(new Tuple(var2, Integer.valueOf(0)));
       int var5 = 0;
 
       while(!var3.isEmpty()) {
          Tuple var6 = (Tuple)var3.poll();
