--- minecraft/src/net/minecraft/block/BlockRedstoneWire.java
+++ src/main/java/net/minecraft/block/BlockRedstoneWire.java
@@ -120,11 +120,11 @@
       return var1.getType(var2.down()).q() || var1.getType(var2.down()).getBlock() == Blocks.GLOWSTONE;
    }
 
    private IBlockState e(World var1, BlockPos var2, IBlockState var3) {
       var3 = this.a(var1, var2, var2, var3);
-      ArrayList var4 = Lists.newArrayList(this.B);
+      ArrayList<BlockPos> var4 = Lists.newArrayList(this.B);
       this.B.clear();
 
       for(BlockPos var6 : var4) {
          var1.applyPhysics(var6, this);
       }
