--- minecraft/src/net/minecraft/block/BlockChorusPlant.java
+++ src/main/java/net/minecraft/block/BlockChorusPlant.java
@@ -55,11 +55,11 @@
       float var10 = ((Boolean)var1.get(c)).booleanValue()?1.0F:0.8125F;
       return new AxisAlignedBB((double)var5, (double)var6, (double)var7, (double)var8, (double)var9, (double)var10);
    }
 
    public void a(IBlockState var1, World var2, BlockPos var3, AxisAlignedBB var4, List<AxisAlignedBB> var5, @Nullable Entity var6) {
-      var1 = var1.b(var2, var3);
+      var1 = var1.b((IBlockAccess)var2, var3);
       float var7 = 0.1875F;
       float var8 = 0.8125F;
       a(var3, var4, var5, new AxisAlignedBB(0.1875D, 0.1875D, 0.1875D, 0.8125D, 0.8125D, 0.8125D));
       if(((Boolean)var1.get(d)).booleanValue()) {
          a(var3, var4, var5, new AxisAlignedBB(0.0D, 0.1875D, 0.1875D, 0.1875D, 0.8125D, 0.8125D));
