--- minecraft/src/net/minecraft/block/BlockPressurePlate.java
+++ src/main/java/net/minecraft/block/BlockPressurePlate.java
@@ -52,11 +52,11 @@
 
    }
 
    protected int e(World var1, BlockPos var2) {
       AxisAlignedBB var3 = c.a(var2);
-      List var4;
+      List<Entity> var4;
       switch(this.e) {
       case EVERYTHING:
          var4 = var1.getEntities((Entity)null, var3);
          break;
       case MOBS:
