--- minecraft/src/net/minecraft/block/BlockPressurePlate.java
+++ src/main/java/net/minecraft/block/BlockPressurePlate.java
@@ -52,11 +52,11 @@
 
    }
 
    protected int computeRedstoneStrength(World var1, BlockPos var2) {
       AxisAlignedBB var3 = PRESSURE_AABB.offset(var2);
-      List var4;
+      List<Entity> var4;
       switch(this.sensitivity) {
       case EVERYTHING:
          var4 = var1.getEntitiesWithinAABBExcludingEntity((Entity)null, var3);
          break;
       case MOBS:
