--- minecraft/src/net/minecraft/world/storage/loot/LootTable.java
+++ src/main/java/net/minecraft/world/storage/loot/LootTable.java
@@ -46,12 +46,12 @@
 
       return var3;
    }
 
    public void fillInventory(IInventory var1, Random var2, LootContext var3) {
-      List var4 = this.generateLootForPools(var2, var3);
-      List var5 = this.getEmptySlotsRandomized(var1, var2);
+      List<ItemStack> var4 = this.generateLootForPools(var2, var3);
+      List<Integer> var5 = this.getEmptySlotsRandomized(var1, var2);
       this.shuffleItems(var4, var5.size(), var2);
 
       for(ItemStack var7 : var4) {
          if(var5.isEmpty()) {
             LOGGER.warn("Tried to over-fill a container");
