--- minecraft/src/net/minecraft/world/storage/loot/LootPool.java
+++ src/main/java/net/minecraft/world/storage/loot/LootPool.java
@@ -34,11 +34,11 @@
       this.rolls = var3;
       this.bonusRolls = var4;
    }
 
    protected void createLootRoll(Collection<ItemStack> var1, Random var2, LootContext var3) {
-      ArrayList var4 = Lists.newArrayList();
+      ArrayList<LootEntry> var4 = Lists.newArrayList();
       int var5 = 0;
 
       for(LootEntry var9 : this.lootEntries) {
          if(LootConditionManager.testAllConditions(var9.conditions, var2, var3)) {
             int var10 = var9.getEffectiveWeight(var3.getLuck());
