--- minecraft/src/net/minecraft/item/crafting/ShapelessRecipes.java
+++ src/main/java/net/minecraft/item/crafting/ShapelessRecipes.java
@@ -35,11 +35,11 @@
 
       return var2;
    }
 
    public boolean a(InventoryCrafting var1, World var2) {
-      ArrayList var3 = Lists.newArrayList(this.ingredients);
+      ArrayList<ItemStack> var3 = Lists.newArrayList(this.ingredients);
 
       for(int var4 = 0; var4 < var1.h(); ++var4) {
          for(int var5 = 0; var5 < var1.i(); ++var5) {
             ItemStack var6 = var1.c(var5, var4);
             if(var6 != null) {
