--- minecraft/src/net/minecraft/util/ClassInheritanceMultiMap.java
+++ src/main/java/net/minecraft/util/ClassInheritanceMultiMap.java
@@ -29,11 +29,11 @@
    }
 
    protected void createLookup(Class<?> var1) {
       ALL_KNOWN.add(var1);
 
-      for(Object var3 : this.values) {
+      for(T var3 : this.values) {
          if(var1.isAssignableFrom(var3.getClass())) {
             this.addForClass(var3, var1);
          }
       }
 
@@ -61,13 +61,13 @@
 
       return true;
    }
 
    private void addForClass(T var1, Class<?> var2) {
-      List var3 = (List)this.map.get(var2);
+      List<T> var3 = this.map.get(var2);
       if(var3 == null) {
-         this.map.put(var2, Lists.newArrayList(new Object[]{var1}));
+         this.map.put(var2, Lists.newArrayList((T)new Object[]{var1}));
       } else {
          var3.add(var1);
       }
 
    }
