--- minecraft/src/net/minecraft/util/IntHashMap.java
+++ src/main/java/net/minecraft/util/IntHashMap.java
@@ -22,11 +22,11 @@
 
    @Nullable
    public V get(int var1) {
       int var2 = g(var1);
 
-      for(IntHashMap.IntHashMapEntry var3 = this.a[a(var2, this.a.length)]; var3 != null; var3 = var3.c) {
+      for(IntHashMap.IntHashMapEntry<V> var3 = this.a[a(var2, this.a.length)]; var3 != null; var3 = var3.c) {
          if(var3.a == var1) {
             return var3.b;
          }
       }
 
