--- minecraft/src/net/minecraft/util/LowerStringMap.java
+++ src/main/java/net/minecraft/util/LowerStringMap.java
@@ -39,11 +39,11 @@
    public V remove(Object var1) {
       return (V)this.internalMap.remove(var1.toString().toLowerCase());
    }
 
    public void putAll(Map<? extends String, ? extends V> var1) {
-      for(Entry var3 : var1.entrySet()) {
+      for(Entry<? extends String, ? extends V> var3 : var1.entrySet()) {
          this.put((String)var3.getKey(), var3.getValue());
       }
 
    }
 
