--- minecraft/src/net/minecraft/util/ObjectIntIdentityMap.java
+++ src/main/java/net/minecraft/util/ObjectIntIdentityMap.java
@@ -24,11 +24,11 @@
 
    public void put(T var1, int var2) {
       this.identityMap.put(var1, Integer.valueOf(var2));
 
       while(this.objectList.size() <= var2) {
-         this.objectList.add((Object)null);
+         this.objectList.add(null);
       }
 
       this.objectList.set(var2, var1);
    }
 
