--- minecraft/src/net/minecraft/network/EnumConnectionState.java
+++ src/main/java/net/minecraft/network/EnumConnectionState.java
@@ -263,11 +263,11 @@
       this.h = Maps.newEnumMap(EnumPacketDirection.class);
       this.g = var3;
    }
 
    protected EnumConnectionState a(EnumPacketDirection var1, Class<? extends Packet<?>> var2) {
-      Object var3 = (BiMap)this.h.get(var1);
+      BiMap<Integer, Class<? extends Packet<?>>> var3 = this.h.get(var1);
       if(var3 == null) {
          var3 = HashBiMap.create();
          this.h.put(var1, var3);
       }
 
@@ -311,11 +311,11 @@
          }
 
          e[var4 - -1] = var3;
 
          for(EnumPacketDirection var6 : var3.h.keySet()) {
-            for(Class var8 : ((BiMap)var3.h.get(var6)).values()) {
+            for(Class var8 : var3.h.get(var6).values()) {
                if(f.containsKey(var8) && f.get(var8) != var3) {
                   throw new Error("Packet " + var8 + " is already assigned to protocol " + f.get(var8) + " - can\'t reassign to " + var3);
                }
 
                try {
