--- minecraft/src/net/minecraft/network/NettyCompressionEncoder.java
+++ src/main/java/net/minecraft/network/NettyCompressionEncoder.java
@@ -14,11 +14,11 @@
    public NettyCompressionEncoder(int var1) {
       this.c = var1;
       this.b = new Deflater();
    }
 
-   protected void a(ChannelHandlerContext var1, ByteBuf var2, ByteBuf var3) throws Exception {
+   protected void encode(ChannelHandlerContext var1, ByteBuf var2, ByteBuf var3) throws Exception {
       int var4 = var2.readableBytes();
       PacketBuffer var5 = new PacketBuffer(var3);
       if(var4 < this.c) {
          var5.d(0);
          var5.writeBytes(var2);
