--- minecraft/src/net/minecraft/network/NetworkManager.java
+++ src/main/java/net/minecraft/network/NetworkManager.java
@@ -110,14 +110,14 @@
 
       g.debug(var2);
       this.close(var3);
    }
 
-   protected void a(ChannelHandlerContext var1, Packet<?> var2) throws Exception {
+   protected void channelRead0(ChannelHandlerContext var1, Packet<?> var2) throws Exception {
       if(this.channel.isOpen()) {
          try {
-            var2.a(this.m);
+            var2.a((PacketBuffer) this.m);
          } catch (ThreadQuickExitException var4) {
             ;
          }
       }
 
