--- minecraft/src/net/minecraft/network/ServerStatusResponse.java
+++ src/main/java/net/minecraft/network/ServerStatusResponse.java
@@ -79,12 +79,12 @@
 
       public void setPlayers(GameProfile[] var1) {
          this.players = var1;
       }
 
-      public static class a implements JsonDeserializer<ServerStatusResponse.a>, JsonSerializer<ServerStatusResponse.a> {
-         public a() {
+      public static class d implements JsonDeserializer<ServerStatusResponse.a>, JsonSerializer<ServerStatusResponse.a> {
+         public d() {
          }
 
          public ServerStatusResponse.a deserialize(JsonElement var1, Type var2, JsonDeserializationContext var3) throws JsonParseException {
             JsonObject var4 = JsonUtils.getJsonObject(var1, "players");
             ServerStatusResponse.a var5 = new ServerStatusResponse.a(JsonUtils.getInt(var4, "max"), JsonUtils.getInt(var4, "online"));
