--- minecraft/src/net/minecraft/item/ItemSlab.java
+++ src/main/java/net/minecraft/item/ItemSlab.java
@@ -81,9 +81,9 @@
       }
 
       return false;
    }
 
-   protected <T extends Comparable<T>> IBlockState a(IProperty<T> var1, Comparable<?> var2) {
-      return this.c.getBlockData().set(var1, var2);
+   protected <T extends Comparable<T>> IBlockState a(IProperty<T> var1, Comparable<T> var2) {
+      return this.c.getBlockData().set(var1, (T) var2);
    }
 }
