--- minecraft/src/net/minecraft/profiler/Profiler.java
+++ src/main/java/net/minecraft/profiler/Profiler.java
@@ -128,10 +128,10 @@
          this.c = var1;
          this.a = var2;
          this.b = var4;
       }
 
-      public int a(Profiler.ProfilerInfo var1) {
+      public int compareTo(Profiler.ProfilerInfo var1) {
          return var1.a < this.a?-1:(var1.a > this.a?1:var1.c.compareTo(this.c));
       }
    }
 }
