--- minecraft/src/net/minecraft/profiler/Snooper.java
+++ src/main/java/net/minecraft/profiler/Snooper.java
@@ -76,11 +76,11 @@
       this.e.b(this);
    }
 
    private void i() {
       RuntimeMXBean var1 = ManagementFactory.getRuntimeMXBean();
-      List var2 = var1.getInputArguments();
+      List<String> var2 = var1.getInputArguments();
       int var3 = 0;
 
       for(String var5 : var2) {
          if(var5.startsWith("-X")) {
             this.a("jvm_arg[" + var3++ + "]", var5);
