--- minecraft/src/net/minecraft/util/math/Cartesian.java
+++ src/main/java/net/minecraft/util/math/Cartesian.java
@@ -43,11 +43,11 @@
    static class a<T> implements Function<Object[], List<T>> {
       private a() {
       }
 
       public List<T> apply(@Nullable Object[] var1) {
-         return Arrays.asList((Object[])var1);
+         return Arrays.asList((T[])var1);
       }
    }
 
    static class b<T> implements Iterable<T[]> {
       private final Class<T> clazz;
