--- minecraft/src/net/minecraft/command/CommandStats.java
+++ src/main/java/net/minecraft/command/CommandStats.java
@@ -143,12 +143,12 @@
    public List<String> getTabCompletionOptions(MinecraftServer var1, ICommandSender var2, String[] var3, @Nullable BlockPos var4) {
       return var3.length == 1?getListOfStringsMatchingLastWord(var3, new String[]{"entity", "block"}):(var3.length == 2 && var3[0].equals("entity")?getListOfStringsMatchingLastWord(var3, var1.getAllUsernames()):(var3.length >= 2 && var3.length <= 4 && var3[0].equals("block")?getTabCompletionCoordinate(var3, 1, var4):((var3.length != 3 || !var3[0].equals("entity")) && (var3.length != 5 || !var3[0].equals("block"))?((var3.length != 4 || !var3[0].equals("entity")) && (var3.length != 6 || !var3[0].equals("block"))?((var3.length != 6 || !var3[0].equals("entity")) && (var3.length != 8 || !var3[0].equals("block"))?Collections.emptyList():getListOfStringsMatchingLastWord(var3, this.getObjectiveNames(var1))):getListOfStringsMatchingLastWord(var3, CommandResultStats.a.getTypeNames())):getListOfStringsMatchingLastWord(var3, new String[]{"set", "clear"}))));
    }
 
    protected List<String> getObjectiveNames(MinecraftServer var1) {
-      Collection var2 = var1.worldServerForDimension(0).getScoreboard().getScoreObjectives();
-      ArrayList var3 = Lists.newArrayList();
+      Collection<ScoreObjective> var2 = var1.worldServerForDimension(0).getScoreboard().getScoreObjectives();
+      ArrayList<String> var3 = Lists.newArrayList();
 
       for(ScoreObjective var5 : var2) {
          if(!var5.getCriteria().isReadOnly()) {
             var3.add(var5.getName());
          }
