--- minecraft/src/net/minecraft/block/properties/PropertyEnum.java
+++ src/main/java/net/minecraft/block/properties/PropertyEnum.java
@@ -24,11 +24,11 @@
          String var6 = ((IStringSerializable)var5).getName();
          if(this.b.containsKey(var6)) {
             throw new IllegalArgumentException("Multiple values have the same name \'" + var6 + "\'");
          }
 
-         this.b.put(var6, var5);
+         this.b.put(var6, (T) var5);
       }
 
    }
 
    public Collection<T> c() {
